{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>AA_N</th>\n",
       "      <th>AA_f</th>\n",
       "      <th>AB_N</th>\n",
       "      <th>AB_f</th>\n",
       "      <th>BB_N</th>\n",
       "      <th>BB_f</th>\n",
       "      <th>ABorBB_N</th>\n",
       "      <th>ABorBB_f</th>\n",
       "      <th>Allelic_Frequency</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "      <th>Otherinfo14</th>\n",
       "      <th>Otherinfo15</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6:161350101:A:G_G</td>\n",
       "      <td>200596</td>\n",
       "      <td>0.999975</td>\n",
       "      <td>5</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161350101_A_G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>49.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=1.2e-05;AQ=49</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:17:17,0:50:0,51,509:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:26:26,0:50:0,78,779:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr6:161350163:C:T_T</td>\n",
       "      <td>200628</td>\n",
       "      <td>0.999990</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161350163_C_T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>45.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=5e-06;AQ=45</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:17:17,0:50:0,51,509:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:18:18,0:50:0,108,1079:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr6:161350208:C:T_T</td>\n",
       "      <td>200497</td>\n",
       "      <td>0.999377</td>\n",
       "      <td>125</td>\n",
       "      <td>0.000623</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>125</td>\n",
       "      <td>0.000623</td>\n",
       "      <td>0.000312</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161350208_C_T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>47.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=0.000314;AQ=47</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:17:17,0:50:0,51,509:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:18:18,0:50:0,108,1079:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr6:161360089:AT:A_A</td>\n",
       "      <td>199266</td>\n",
       "      <td>0.999970</td>\n",
       "      <td>6</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161360089_AT_A</td>\n",
       "      <td>AT</td>\n",
       "      <td>A</td>\n",
       "      <td>38.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=5e-06;AQ=38</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>./.:14:14,0:0:0,42,419:II</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:14:14,0:42:0,42,419:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr6:161360132:GT:G_G</td>\n",
       "      <td>200627</td>\n",
       "      <td>0.999990</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161360132_GT_G</td>\n",
       "      <td>GT</td>\n",
       "      <td>G</td>\n",
       "      <td>49.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=5e-06;AQ=49</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,57,569:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:16:16,0:48:0,81,809:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr6:161548898:G:A_A</td>\n",
       "      <td>200629</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161548898_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>43.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=43</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,87,869:..</td>\n",
       "      <td>0/0:16:16,0:48:0,54,539:..</td>\n",
       "      <td>0/0:16:16,0:48:0,60,599:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:161548965:GA:G_G</td>\n",
       "      <td>200627</td>\n",
       "      <td>0.999990</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161548965_GA_G</td>\n",
       "      <td>GA</td>\n",
       "      <td>G</td>\n",
       "      <td>39.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=5e-06;AQ=23</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,87,869:..</td>\n",
       "      <td>0/0:16:16,0:48:0,54,539:..</td>\n",
       "      <td>0/0:16:16,0:48:0,60,599:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr6:161548979:C:CACCA_CACCA</td>\n",
       "      <td>200628</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161548979_C_CACCA</td>\n",
       "      <td>C</td>\n",
       "      <td>CACCA</td>\n",
       "      <td>44.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=44</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,87,869:..</td>\n",
       "      <td>0/0:16:16,0:48:0,54,539:..</td>\n",
       "      <td>0/0:16:16,0:48:0,60,599:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr6:161569397:CA:C_C</td>\n",
       "      <td>200589</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161569397_CA_C</td>\n",
       "      <td>CA</td>\n",
       "      <td>C</td>\n",
       "      <td>40.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=37</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:18:18,0:50:0,72,719:..</td>\n",
       "      <td>0/0:20:20,0:50:0,60,599:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr6:161785793:C:G_G</td>\n",
       "      <td>200629</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161785793_C_G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>37.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=37</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:17:17,0:50:0,66,659:..</td>\n",
       "      <td>0/0:20:20,0:50:0,69,689:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr6:161785817:G:A_A</td>\n",
       "      <td>200629</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161785817_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>35.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=35</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:17:17,0:50:0,66,659:..</td>\n",
       "      <td>0/0:20:20,0:50:0,69,689:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr6:161785820:G:A_A</td>\n",
       "      <td>199071</td>\n",
       "      <td>0.992477</td>\n",
       "      <td>1503</td>\n",
       "      <td>0.007493</td>\n",
       "      <td>6</td>\n",
       "      <td>2.99133e-05</td>\n",
       "      <td>1509</td>\n",
       "      <td>0.007523</td>\n",
       "      <td>0.003777</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161785820_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>55.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=0.003878;AQ=55</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:17:17,0:50:0,66,659:..</td>\n",
       "      <td>0/0:20:20,0:50:0,69,689:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr6:161785839:A:T_T</td>\n",
       "      <td>200628</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161785839_A_T</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>44.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=44</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:20:20,0:50:0,69,689:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr6:161785885:C:T_T</td>\n",
       "      <td>200628</td>\n",
       "      <td>0.999990</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161785885_C_T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>41.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=5e-06;AQ=41</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,75,749:..</td>\n",
       "      <td>0/0:20:20,0:50:0,69,689:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr6:161973317:G:A_A</td>\n",
       "      <td>200473</td>\n",
       "      <td>0.999616</td>\n",
       "      <td>77</td>\n",
       "      <td>0.000384</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>77</td>\n",
       "      <td>0.000384</td>\n",
       "      <td>0.000192</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161973317_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>51.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=0.000194;AQ=51</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:16:16,0:48:0,63,629:..</td>\n",
       "      <td>0/0:22:22,0:50:0,66,659:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr6:161973323:C:CAAGT_CAAGT</td>\n",
       "      <td>200216</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161973323_C_CAAGT</td>\n",
       "      <td>C</td>\n",
       "      <td>CAAGT</td>\n",
       "      <td>44.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=44</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:16:16,0:48:0,63,629:..</td>\n",
       "      <td>0/0:22:22,0:50:0,66,659:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr6:161973358:CA:C_C</td>\n",
       "      <td>200607</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161973358_CA_C</td>\n",
       "      <td>CA</td>\n",
       "      <td>C</td>\n",
       "      <td>40.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=36</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:16:16,0:48:0,63,629:..</td>\n",
       "      <td>0/0:22:22,0:50:0,66,659:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr6:161973401:C:T_T</td>\n",
       "      <td>200580</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161973401_C_T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>42.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=42</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:16:16,0:48:0,63,629:..</td>\n",
       "      <td>0/0:22:22,0:50:0,66,659:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr6:161973403:T:A_A</td>\n",
       "      <td>200553</td>\n",
       "      <td>0.999965</td>\n",
       "      <td>7</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161973403_T_A</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>45.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=1.7e-05;AQ=45</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:16:16,0:48:0,63,629:..</td>\n",
       "      <td>0/0:22:22,0:50:0,66,659:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr6:161973414:AT:A_A</td>\n",
       "      <td>194282</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161973414_AT_A</td>\n",
       "      <td>AT</td>\n",
       "      <td>A</td>\n",
       "      <td>42.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=42</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:16:16,0:48:0,63,629:..</td>\n",
       "      <td>0/0:18:18,0:24:0,24,479:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr6:161973414:AT:ATT_ATT</td>\n",
       "      <td>194282</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161973414_AT_ATT</td>\n",
       "      <td>AT</td>\n",
       "      <td>ATT</td>\n",
       "      <td>42.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=42</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:16:16,0:48:0,63,629:..</td>\n",
       "      <td>0/0:18:18,0:24:0,24,479:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr6:162201236:G:T_T</td>\n",
       "      <td>200616</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162201236_G_T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>45.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=45</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr6:162262537:C:CTGG_CTGG</td>\n",
       "      <td>200623</td>\n",
       "      <td>0.999965</td>\n",
       "      <td>7</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162262537_C_CTGG</td>\n",
       "      <td>C</td>\n",
       "      <td>CTGG</td>\n",
       "      <td>48.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=1.7e-05;AQ=48</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:21:21,0:48:0,66,659:..</td>\n",
       "      <td>0/0:19:19,0:50:0,60,599:..</td>\n",
       "      <td>0/0:33:33,0:50:0,99,989:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr6:162262543:G:GT_GT</td>\n",
       "      <td>200623</td>\n",
       "      <td>0.999965</td>\n",
       "      <td>7</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162262543_G_GT</td>\n",
       "      <td>G</td>\n",
       "      <td>GT</td>\n",
       "      <td>46.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=1.7e-05;AQ=46</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:21:21,0:48:0,66,659:..</td>\n",
       "      <td>0/0:19:19,0:50:0,60,599:..</td>\n",
       "      <td>0/0:33:33,0:50:0,99,989:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr6:162262560:TCAGTGTGCAGAATGACAGCCAGCCCCACAG...</td>\n",
       "      <td>200473</td>\n",
       "      <td>0.999342</td>\n",
       "      <td>132</td>\n",
       "      <td>0.000658</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>132</td>\n",
       "      <td>0.000658</td>\n",
       "      <td>0.000329</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162262560_TCAGTGTGCAGAATGACAGCCAGCCCCACAG...</td>\n",
       "      <td>TCAGTGTGCAGAATGACAGCCAGCCCCACAGAGTCTCCTGG</td>\n",
       "      <td>T</td>\n",
       "      <td>50.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=0.000202;AQ=38</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:21:21,0:48:0,66,659:..</td>\n",
       "      <td>0/0:19:19,0:50:0,60,599:..</td>\n",
       "      <td>0/0:33:33,0:50:0,99,989:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr6:162262560:TCAGTGTGCAGAATGACAGCCAGCCCCACAG...</td>\n",
       "      <td>200604</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162262560_TCAGTGTGCAGAATGACAGCCAGCCCCACAG...</td>\n",
       "      <td>TCAGTGTGCAGAATGACAGCCAGCCCCACAGAGTCTCCTGG</td>\n",
       "      <td>TCAGGTGTGCAGAATGACAGCCAGCCCCACAGAGTCTCCTGG</td>\n",
       "      <td>50.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=40</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:21:21,0:48:0,66,659:..</td>\n",
       "      <td>0/0:19:19,0:50:0,60,599:..</td>\n",
       "      <td>0/0:33:33,0:50:0,99,989:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr6:162262662:G:GAA_GAA</td>\n",
       "      <td>200628</td>\n",
       "      <td>0.999990</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162262662_G_GAA</td>\n",
       "      <td>G</td>\n",
       "      <td>GAA</td>\n",
       "      <td>39.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=5e-06;AQ=29</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:21:21,0:48:0,66,659:..</td>\n",
       "      <td>0/0:19:19,0:50:0,60,599:..</td>\n",
       "      <td>0/0:33:33,0:50:0,99,989:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr6:162262702:C:A_A</td>\n",
       "      <td>200623</td>\n",
       "      <td>0.999965</td>\n",
       "      <td>7</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162262702_C_A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>50.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=1.7e-05;AQ=50</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:21:21,0:48:0,66,659:..</td>\n",
       "      <td>0/0:19:19,0:50:0,60,599:..</td>\n",
       "      <td>0/0:33:33,0:50:0,99,989:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr6:162262715:C:CCA_CCA</td>\n",
       "      <td>200620</td>\n",
       "      <td>0.999950</td>\n",
       "      <td>10</td>\n",
       "      <td>0.000050</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0.000050</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162262715_C_CCA</td>\n",
       "      <td>C</td>\n",
       "      <td>CCA</td>\n",
       "      <td>43.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2.5e-05;AQ=43</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:21:21,0:48:0,66,659:..</td>\n",
       "      <td>0/0:19:19,0:50:0,60,599:..</td>\n",
       "      <td>0/0:33:33,0:50:0,99,989:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr6:162443325:AT:A_A</td>\n",
       "      <td>200501</td>\n",
       "      <td>0.999377</td>\n",
       "      <td>125</td>\n",
       "      <td>0.000623</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>125</td>\n",
       "      <td>0.000623</td>\n",
       "      <td>0.000312</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162443325_AT_A</td>\n",
       "      <td>AT</td>\n",
       "      <td>A</td>\n",
       "      <td>56.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=0.000314;AQ=56</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:30:30,0:50:0,90,899:..</td>\n",
       "      <td>0/0:16:16,0:48:0,60,599:..</td>\n",
       "      <td>0/0:17:17,0:50:0,84,839:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr6:162443356:C:G_G</td>\n",
       "      <td>200612</td>\n",
       "      <td>0.999915</td>\n",
       "      <td>17</td>\n",
       "      <td>0.000085</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>0.000085</td>\n",
       "      <td>0.000042</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162443356_C_G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>49.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=4.5e-05;AQ=43</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:34:34,0:48:0,123,1229:..</td>\n",
       "      <td>0/0:16:16,0:48:0,60,599:..</td>\n",
       "      <td>0/0:17:17,0:50:0,84,839:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr6:162443378:CCT:C_C</td>\n",
       "      <td>200465</td>\n",
       "      <td>0.999217</td>\n",
       "      <td>157</td>\n",
       "      <td>0.000783</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>157</td>\n",
       "      <td>0.000783</td>\n",
       "      <td>0.000391</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162443378_CCT_C</td>\n",
       "      <td>CCT</td>\n",
       "      <td>C</td>\n",
       "      <td>55.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=0.000409;AQ=55</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:34:34,0:48:0,123,1229:..</td>\n",
       "      <td>0/0:16:16,0:48:0,60,599:..</td>\n",
       "      <td>0/0:17:17,0:50:0,84,839:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr6:162443378:CCT:CCC_CCC</td>\n",
       "      <td>200380</td>\n",
       "      <td>0.998784</td>\n",
       "      <td>243</td>\n",
       "      <td>0.001211</td>\n",
       "      <td>1</td>\n",
       "      <td>4.98445e-06</td>\n",
       "      <td>244</td>\n",
       "      <td>0.001216</td>\n",
       "      <td>0.000611</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162443378_CCT_CCC</td>\n",
       "      <td>CCT</td>\n",
       "      <td>CCC</td>\n",
       "      <td>55.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=0.000635;AQ=53</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:34:34,0:48:0,123,1229:..</td>\n",
       "      <td>0/0:16:16,0:48:0,60,599:..</td>\n",
       "      <td>0/0:17:17,0:50:0,84,839:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr6:162443383:C:T_T</td>\n",
       "      <td>200534</td>\n",
       "      <td>0.999526</td>\n",
       "      <td>95</td>\n",
       "      <td>0.000474</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>95</td>\n",
       "      <td>0.000474</td>\n",
       "      <td>0.000237</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162443383_C_T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>50.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=0.000239;AQ=50</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:34:34,0:48:0,123,1229:..</td>\n",
       "      <td>0/0:16:16,0:48:0,60,599:..</td>\n",
       "      <td>0/0:17:17,0:50:0,84,839:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr6:162443384:G:A_A</td>\n",
       "      <td>200629</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162443384_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>35.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=35</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:34:34,0:48:0,123,1229:..</td>\n",
       "      <td>0/0:16:16,0:48:0,60,599:..</td>\n",
       "      <td>0/0:17:17,0:50:0,84,839:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr6:162727667:A:G_G</td>\n",
       "      <td>200619</td>\n",
       "      <td>0.999980</td>\n",
       "      <td>4</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162727667_A_G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>42.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=1e-05;AQ=42</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:20:20,0:30:0,30,539:..</td>\n",
       "      <td>0/0:16:16,0:48:0,66,659:..</td>\n",
       "      <td>0/0:20:20,0:50:0,66,659:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr6:162727668:T:C_C</td>\n",
       "      <td>200626</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162727668_T_C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>45.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=45</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:17:17,0:50:0,63,629:..</td>\n",
       "      <td>0/0:16:16,0:48:0,66,659:..</td>\n",
       "      <td>0/0:20:20,0:50:0,66,659:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>WESTotal</td>\n",
       "      <td>198076</td>\n",
       "      <td>0.987270</td>\n",
       "      <td>2542</td>\n",
       "      <td>0.012670</td>\n",
       "      <td>12</td>\n",
       "      <td>5.98116e-05</td>\n",
       "      <td>2554</td>\n",
       "      <td>0.012730</td>\n",
       "      <td>0.006395</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>DEL1</td>\n",
       "      <td>199981</td>\n",
       "      <td>0.996765</td>\n",
       "      <td>649</td>\n",
       "      <td>0.003235</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>649</td>\n",
       "      <td>0.003235</td>\n",
       "      <td>0.001617</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>DEL1_Exon</td>\n",
       "      <td>199981</td>\n",
       "      <td>0.996765</td>\n",
       "      <td>244</td>\n",
       "      <td>0.001216</td>\n",
       "      <td>94</td>\n",
       "      <td>0.000468524</td>\n",
       "      <td>338</td>\n",
       "      <td>0.001685</td>\n",
       "      <td>0.001077</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>DUP1</td>\n",
       "      <td>200130</td>\n",
       "      <td>0.997508</td>\n",
       "      <td>500</td>\n",
       "      <td>0.002492</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>500</td>\n",
       "      <td>0.002492</td>\n",
       "      <td>0.001246</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>DUP1_Exon</td>\n",
       "      <td>200130</td>\n",
       "      <td>0.997508</td>\n",
       "      <td>320</td>\n",
       "      <td>0.001595</td>\n",
       "      <td>58</td>\n",
       "      <td>0.000289089</td>\n",
       "      <td>378</td>\n",
       "      <td>0.001884</td>\n",
       "      <td>0.001087</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>2DELDUP</td>\n",
       "      <td>200628</td>\n",
       "      <td>0.999990</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>9.9686e-06</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>CNVTotal</td>\n",
       "      <td>199479</td>\n",
       "      <td>0.994263</td>\n",
       "      <td>1149</td>\n",
       "      <td>0.005727</td>\n",
       "      <td>2</td>\n",
       "      <td>9.9686e-06</td>\n",
       "      <td>1151</td>\n",
       "      <td>0.005737</td>\n",
       "      <td>0.002873</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>MutTotal</td>\n",
       "      <td>196935</td>\n",
       "      <td>0.981583</td>\n",
       "      <td>3671</td>\n",
       "      <td>0.018297</td>\n",
       "      <td>24</td>\n",
       "      <td>0.000119623</td>\n",
       "      <td>3695</td>\n",
       "      <td>0.018417</td>\n",
       "      <td>0.009268</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>45 rows Ã— 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   ID    AA_N      AA_f  AB_N  \\\n",
       "0                                chr6:161350101:A:G_G  200596  0.999975     5   \n",
       "1                                chr6:161350163:C:T_T  200628  0.999990     2   \n",
       "2                                chr6:161350208:C:T_T  200497  0.999377   125   \n",
       "3                               chr6:161360089:AT:A_A  199266  0.999970     6   \n",
       "4                               chr6:161360132:GT:G_G  200627  0.999990     2   \n",
       "5                                chr6:161548898:G:A_A  200629  0.999995     1   \n",
       "6                               chr6:161548965:GA:G_G  200627  0.999990     2   \n",
       "7                        chr6:161548979:C:CACCA_CACCA  200628  0.999995     1   \n",
       "8                               chr6:161569397:CA:C_C  200589  0.999995     1   \n",
       "9                                chr6:161785793:C:G_G  200629  0.999995     1   \n",
       "10                               chr6:161785817:G:A_A  200629  0.999995     1   \n",
       "11                               chr6:161785820:G:A_A  199071  0.992477  1503   \n",
       "12                               chr6:161785839:A:T_T  200628  0.999995     1   \n",
       "13                               chr6:161785885:C:T_T  200628  0.999990     2   \n",
       "14                               chr6:161973317:G:A_A  200473  0.999616    77   \n",
       "15                       chr6:161973323:C:CAAGT_CAAGT  200216  0.999995     1   \n",
       "16                              chr6:161973358:CA:C_C  200607  0.999995     1   \n",
       "17                               chr6:161973401:C:T_T  200580  0.999995     1   \n",
       "18                               chr6:161973403:T:A_A  200553  0.999965     7   \n",
       "19                              chr6:161973414:AT:A_A  194282  0.999995     1   \n",
       "20                          chr6:161973414:AT:ATT_ATT  194282  0.999995     1   \n",
       "21                               chr6:162201236:G:T_T  200616  0.999995     1   \n",
       "22                         chr6:162262537:C:CTGG_CTGG  200623  0.999965     7   \n",
       "23                             chr6:162262543:G:GT_GT  200623  0.999965     7   \n",
       "24  chr6:162262560:TCAGTGTGCAGAATGACAGCCAGCCCCACAG...  200473  0.999342   132   \n",
       "25  chr6:162262560:TCAGTGTGCAGAATGACAGCCAGCCCCACAG...  200604  0.999995     1   \n",
       "26                           chr6:162262662:G:GAA_GAA  200628  0.999990     2   \n",
       "27                               chr6:162262702:C:A_A  200623  0.999965     7   \n",
       "28                           chr6:162262715:C:CCA_CCA  200620  0.999950    10   \n",
       "29                              chr6:162443325:AT:A_A  200501  0.999377   125   \n",
       "30                               chr6:162443356:C:G_G  200612  0.999915    17   \n",
       "31                             chr6:162443378:CCT:C_C  200465  0.999217   157   \n",
       "32                         chr6:162443378:CCT:CCC_CCC  200380  0.998784   243   \n",
       "33                               chr6:162443383:C:T_T  200534  0.999526    95   \n",
       "34                               chr6:162443384:G:A_A  200629  0.999995     1   \n",
       "35                               chr6:162727667:A:G_G  200619  0.999980     4   \n",
       "36                               chr6:162727668:T:C_C  200626  0.999995     1   \n",
       "37                                           WESTotal  198076  0.987270  2542   \n",
       "38                                               DEL1  199981  0.996765   649   \n",
       "39                                          DEL1_Exon  199981  0.996765   244   \n",
       "40                                               DUP1  200130  0.997508   500   \n",
       "41                                          DUP1_Exon  200130  0.997508   320   \n",
       "42                                            2DELDUP  200628  0.999990     0   \n",
       "43                                           CNVTotal  199479  0.994263  1149   \n",
       "44                                           MutTotal  196935  0.981583  3671   \n",
       "\n",
       "        AB_f BB_N         BB_f ABorBB_N  ABorBB_f  Allelic_Frequency  ...  \\\n",
       "0   0.000025    0            0        5  0.000025           0.000012  ...   \n",
       "1   0.000010    0            0        2  0.000010           0.000005  ...   \n",
       "2   0.000623    0            0      125  0.000623           0.000312  ...   \n",
       "3   0.000030    0            0        6  0.000030           0.000015  ...   \n",
       "4   0.000010    0            0        2  0.000010           0.000005  ...   \n",
       "5   0.000005    0            0        1  0.000005           0.000002  ...   \n",
       "6   0.000010    0            0        2  0.000010           0.000005  ...   \n",
       "7   0.000005    0            0        1  0.000005           0.000002  ...   \n",
       "8   0.000005    0            0        1  0.000005           0.000002  ...   \n",
       "9   0.000005    0            0        1  0.000005           0.000002  ...   \n",
       "10  0.000005    0            0        1  0.000005           0.000002  ...   \n",
       "11  0.007493    6  2.99133e-05     1509  0.007523           0.003777  ...   \n",
       "12  0.000005    0            0        1  0.000005           0.000002  ...   \n",
       "13  0.000010    0            0        2  0.000010           0.000005  ...   \n",
       "14  0.000384    0            0       77  0.000384           0.000192  ...   \n",
       "15  0.000005    0            0        1  0.000005           0.000002  ...   \n",
       "16  0.000005    0            0        1  0.000005           0.000002  ...   \n",
       "17  0.000005    0            0        1  0.000005           0.000002  ...   \n",
       "18  0.000035    0            0        7  0.000035           0.000017  ...   \n",
       "19  0.000005    0            0        1  0.000005           0.000003  ...   \n",
       "20  0.000005    0            0        1  0.000005           0.000003  ...   \n",
       "21  0.000005    0            0        1  0.000005           0.000002  ...   \n",
       "22  0.000035    0            0        7  0.000035           0.000017  ...   \n",
       "23  0.000035    0            0        7  0.000035           0.000017  ...   \n",
       "24  0.000658    0            0      132  0.000658           0.000329  ...   \n",
       "25  0.000005    0            0        1  0.000005           0.000002  ...   \n",
       "26  0.000010    0            0        2  0.000010           0.000005  ...   \n",
       "27  0.000035    0            0        7  0.000035           0.000017  ...   \n",
       "28  0.000050    0            0       10  0.000050           0.000025  ...   \n",
       "29  0.000623    0            0      125  0.000623           0.000312  ...   \n",
       "30  0.000085    0            0       17  0.000085           0.000042  ...   \n",
       "31  0.000783    0            0      157  0.000783           0.000391  ...   \n",
       "32  0.001211    1  4.98445e-06      244  0.001216           0.000611  ...   \n",
       "33  0.000474    0            0       95  0.000474           0.000237  ...   \n",
       "34  0.000005    0            0        1  0.000005           0.000002  ...   \n",
       "35  0.000020    0            0        4  0.000020           0.000010  ...   \n",
       "36  0.000005    0            0        1  0.000005           0.000002  ...   \n",
       "37  0.012670   12  5.98116e-05     2554  0.012730           0.006395  ...   \n",
       "38  0.003235    0            0      649  0.003235           0.001617  ...   \n",
       "39  0.001216   94  0.000468524      338  0.001685           0.001077  ...   \n",
       "40  0.002492    0            0      500  0.002492           0.001246  ...   \n",
       "41  0.001595   58  0.000289089      378  0.001884           0.001087  ...   \n",
       "42  0.000000    2   9.9686e-06        2  0.000010           0.000010  ...   \n",
       "43  0.005727    2   9.9686e-06     1151  0.005737           0.002873  ...   \n",
       "44  0.018297   24  0.000119623     3695  0.018417           0.009268  ...   \n",
       "\n",
       "                                           Otherinfo6  \\\n",
       "0                                  chr6_161350101_A_G   \n",
       "1                                  chr6_161350163_C_T   \n",
       "2                                  chr6_161350208_C_T   \n",
       "3                                 chr6_161360089_AT_A   \n",
       "4                                 chr6_161360132_GT_G   \n",
       "5                                  chr6_161548898_G_A   \n",
       "6                                 chr6_161548965_GA_G   \n",
       "7                              chr6_161548979_C_CACCA   \n",
       "8                                 chr6_161569397_CA_C   \n",
       "9                                  chr6_161785793_C_G   \n",
       "10                                 chr6_161785817_G_A   \n",
       "11                                 chr6_161785820_G_A   \n",
       "12                                 chr6_161785839_A_T   \n",
       "13                                 chr6_161785885_C_T   \n",
       "14                                 chr6_161973317_G_A   \n",
       "15                             chr6_161973323_C_CAAGT   \n",
       "16                                chr6_161973358_CA_C   \n",
       "17                                 chr6_161973401_C_T   \n",
       "18                                 chr6_161973403_T_A   \n",
       "19                                chr6_161973414_AT_A   \n",
       "20                              chr6_161973414_AT_ATT   \n",
       "21                                 chr6_162201236_G_T   \n",
       "22                              chr6_162262537_C_CTGG   \n",
       "23                                chr6_162262543_G_GT   \n",
       "24  chr6_162262560_TCAGTGTGCAGAATGACAGCCAGCCCCACAG...   \n",
       "25  chr6_162262560_TCAGTGTGCAGAATGACAGCCAGCCCCACAG...   \n",
       "26                               chr6_162262662_G_GAA   \n",
       "27                                 chr6_162262702_C_A   \n",
       "28                               chr6_162262715_C_CCA   \n",
       "29                                chr6_162443325_AT_A   \n",
       "30                                 chr6_162443356_C_G   \n",
       "31                               chr6_162443378_CCT_C   \n",
       "32                             chr6_162443378_CCT_CCC   \n",
       "33                                 chr6_162443383_C_T   \n",
       "34                                 chr6_162443384_G_A   \n",
       "35                                 chr6_162727667_A_G   \n",
       "36                                 chr6_162727668_T_C   \n",
       "37                                                NaN   \n",
       "38                                                NaN   \n",
       "39                                                NaN   \n",
       "40                                                NaN   \n",
       "41                                                NaN   \n",
       "42                                                NaN   \n",
       "43                                                NaN   \n",
       "44                                                NaN   \n",
       "\n",
       "                                   Otherinfo7  \\\n",
       "0                                           A   \n",
       "1                                           C   \n",
       "2                                           C   \n",
       "3                                          AT   \n",
       "4                                          GT   \n",
       "5                                           G   \n",
       "6                                          GA   \n",
       "7                                           C   \n",
       "8                                          CA   \n",
       "9                                           C   \n",
       "10                                          G   \n",
       "11                                          G   \n",
       "12                                          A   \n",
       "13                                          C   \n",
       "14                                          G   \n",
       "15                                          C   \n",
       "16                                         CA   \n",
       "17                                          C   \n",
       "18                                          T   \n",
       "19                                         AT   \n",
       "20                                         AT   \n",
       "21                                          G   \n",
       "22                                          C   \n",
       "23                                          G   \n",
       "24  TCAGTGTGCAGAATGACAGCCAGCCCCACAGAGTCTCCTGG   \n",
       "25  TCAGTGTGCAGAATGACAGCCAGCCCCACAGAGTCTCCTGG   \n",
       "26                                          G   \n",
       "27                                          C   \n",
       "28                                          C   \n",
       "29                                         AT   \n",
       "30                                          C   \n",
       "31                                        CCT   \n",
       "32                                        CCT   \n",
       "33                                          C   \n",
       "34                                          G   \n",
       "35                                          A   \n",
       "36                                          T   \n",
       "37                                        NaN   \n",
       "38                                        NaN   \n",
       "39                                        NaN   \n",
       "40                                        NaN   \n",
       "41                                        NaN   \n",
       "42                                        NaN   \n",
       "43                                        NaN   \n",
       "44                                        NaN   \n",
       "\n",
       "                                    Otherinfo8 Otherinfo9 Otherinfo10  \\\n",
       "0                                            G       49.0           .   \n",
       "1                                            T       45.0           .   \n",
       "2                                            T       47.0           .   \n",
       "3                                            A       38.0           .   \n",
       "4                                            G       49.0           .   \n",
       "5                                            A       43.0           .   \n",
       "6                                            G       39.0           .   \n",
       "7                                        CACCA       44.0           .   \n",
       "8                                            C       40.0           .   \n",
       "9                                            G       37.0           .   \n",
       "10                                           A       35.0           .   \n",
       "11                                           A       55.0           .   \n",
       "12                                           T       44.0           .   \n",
       "13                                           T       41.0           .   \n",
       "14                                           A       51.0           .   \n",
       "15                                       CAAGT       44.0           .   \n",
       "16                                           C       40.0           .   \n",
       "17                                           T       42.0           .   \n",
       "18                                           A       45.0           .   \n",
       "19                                           A       42.0           .   \n",
       "20                                         ATT       42.0           .   \n",
       "21                                           T       45.0           .   \n",
       "22                                        CTGG       48.0           .   \n",
       "23                                          GT       46.0           .   \n",
       "24                                           T       50.0           .   \n",
       "25  TCAGGTGTGCAGAATGACAGCCAGCCCCACAGAGTCTCCTGG       50.0           .   \n",
       "26                                         GAA       39.0           .   \n",
       "27                                           A       50.0           .   \n",
       "28                                         CCA       43.0           .   \n",
       "29                                           A       56.0           .   \n",
       "30                                           G       49.0           .   \n",
       "31                                           C       55.0           .   \n",
       "32                                         CCC       55.0           .   \n",
       "33                                           T       50.0           .   \n",
       "34                                           A       35.0           .   \n",
       "35                                           G       42.0           .   \n",
       "36                                           C       45.0           .   \n",
       "37                                         NaN        NaN         NaN   \n",
       "38                                         NaN        NaN         NaN   \n",
       "39                                         NaN        NaN         NaN   \n",
       "40                                         NaN        NaN         NaN   \n",
       "41                                         NaN        NaN         NaN   \n",
       "42                                         NaN        NaN         NaN   \n",
       "43                                         NaN        NaN         NaN   \n",
       "44                                         NaN        NaN         NaN   \n",
       "\n",
       "          Otherinfo11         Otherinfo12                   Otherinfo13  \\\n",
       "0    AF=1.2e-05;AQ=49  GT:DP:AD:GQ:PL:RNC    0/0:17:17,0:50:0,51,509:..   \n",
       "1      AF=5e-06;AQ=45  GT:DP:AD:GQ:PL:RNC    0/0:17:17,0:50:0,51,509:..   \n",
       "2   AF=0.000314;AQ=47  GT:DP:AD:GQ:PL:RNC    0/0:17:17,0:50:0,51,509:..   \n",
       "3      AF=5e-06;AQ=38  GT:DP:AD:GQ:PL:RNC     ./.:14:14,0:0:0,42,419:II   \n",
       "4      AF=5e-06;AQ=49  GT:DP:AD:GQ:PL:RNC    0/0:16:16,0:48:0,57,569:..   \n",
       "5      AF=2e-06;AQ=43  GT:DP:AD:GQ:PL:RNC    0/0:16:16,0:48:0,87,869:..   \n",
       "6      AF=5e-06;AQ=23  GT:DP:AD:GQ:PL:RNC    0/0:16:16,0:48:0,87,869:..   \n",
       "7      AF=2e-06;AQ=44  GT:DP:AD:GQ:PL:RNC    0/0:16:16,0:48:0,87,869:..   \n",
       "8      AF=2e-06;AQ=37  GT:DP:AD:GQ:PL:RNC    0/0:18:18,0:50:0,72,719:..   \n",
       "9      AF=2e-06;AQ=37  GT:DP:AD:GQ:PL:RNC    0/0:17:17,0:50:0,66,659:..   \n",
       "10     AF=2e-06;AQ=35  GT:DP:AD:GQ:PL:RNC    0/0:17:17,0:50:0,66,659:..   \n",
       "11  AF=0.003878;AQ=55  GT:DP:AD:GQ:PL:RNC    0/0:17:17,0:50:0,66,659:..   \n",
       "12     AF=2e-06;AQ=44  GT:DP:AD:GQ:PL:RNC    0/0:16:16,0:48:0,48,479:..   \n",
       "13     AF=5e-06;AQ=41  GT:DP:AD:GQ:PL:RNC    0/0:16:16,0:48:0,75,749:..   \n",
       "14  AF=0.000194;AQ=51  GT:DP:AD:GQ:PL:RNC    0/0:16:16,0:48:0,48,479:..   \n",
       "15     AF=2e-06;AQ=44  GT:DP:AD:GQ:PL:RNC    0/0:16:16,0:48:0,48,479:..   \n",
       "16     AF=2e-06;AQ=36  GT:DP:AD:GQ:PL:RNC    0/0:16:16,0:48:0,48,479:..   \n",
       "17     AF=2e-06;AQ=42  GT:DP:AD:GQ:PL:RNC    0/0:16:16,0:48:0,48,479:..   \n",
       "18   AF=1.7e-05;AQ=45  GT:DP:AD:GQ:PL:RNC    0/0:16:16,0:48:0,48,479:..   \n",
       "19     AF=2e-06;AQ=42  GT:DP:AD:GQ:PL:RNC    0/0:16:16,0:48:0,48,479:..   \n",
       "20     AF=2e-06;AQ=42  GT:DP:AD:GQ:PL:RNC    0/0:16:16,0:48:0,48,479:..   \n",
       "21     AF=2e-06;AQ=45  GT:DP:AD:GQ:PL:RNC    0/0:16:16,0:48:0,48,479:..   \n",
       "22   AF=1.7e-05;AQ=48  GT:DP:AD:GQ:PL:RNC    0/0:21:21,0:48:0,66,659:..   \n",
       "23   AF=1.7e-05;AQ=46  GT:DP:AD:GQ:PL:RNC    0/0:21:21,0:48:0,66,659:..   \n",
       "24  AF=0.000202;AQ=38  GT:DP:AD:GQ:PL:RNC    0/0:21:21,0:48:0,66,659:..   \n",
       "25     AF=2e-06;AQ=40  GT:DP:AD:GQ:PL:RNC    0/0:21:21,0:48:0,66,659:..   \n",
       "26     AF=5e-06;AQ=29  GT:DP:AD:GQ:PL:RNC    0/0:21:21,0:48:0,66,659:..   \n",
       "27   AF=1.7e-05;AQ=50  GT:DP:AD:GQ:PL:RNC    0/0:21:21,0:48:0,66,659:..   \n",
       "28   AF=2.5e-05;AQ=43  GT:DP:AD:GQ:PL:RNC    0/0:21:21,0:48:0,66,659:..   \n",
       "29  AF=0.000314;AQ=56  GT:DP:AD:GQ:PL:RNC    0/0:30:30,0:50:0,90,899:..   \n",
       "30   AF=4.5e-05;AQ=43  GT:DP:AD:GQ:PL:RNC  0/0:34:34,0:48:0,123,1229:..   \n",
       "31  AF=0.000409;AQ=55  GT:DP:AD:GQ:PL:RNC  0/0:34:34,0:48:0,123,1229:..   \n",
       "32  AF=0.000635;AQ=53  GT:DP:AD:GQ:PL:RNC  0/0:34:34,0:48:0,123,1229:..   \n",
       "33  AF=0.000239;AQ=50  GT:DP:AD:GQ:PL:RNC  0/0:34:34,0:48:0,123,1229:..   \n",
       "34     AF=2e-06;AQ=35  GT:DP:AD:GQ:PL:RNC  0/0:34:34,0:48:0,123,1229:..   \n",
       "35     AF=1e-05;AQ=42  GT:DP:AD:GQ:PL:RNC    0/0:20:20,0:30:0,30,539:..   \n",
       "36     AF=2e-06;AQ=45  GT:DP:AD:GQ:PL:RNC    0/0:17:17,0:50:0,63,629:..   \n",
       "37                NaN                 NaN                           NaN   \n",
       "38                NaN                 NaN                           NaN   \n",
       "39                NaN                 NaN                           NaN   \n",
       "40                NaN                 NaN                           NaN   \n",
       "41                NaN                 NaN                           NaN   \n",
       "42                NaN                 NaN                           NaN   \n",
       "43                NaN                 NaN                           NaN   \n",
       "44                NaN                 NaN                           NaN   \n",
       "\n",
       "                   Otherinfo14                   Otherinfo15  \n",
       "0   0/0:16:16,0:48:0,48,479:..    0/0:26:26,0:50:0,78,779:..  \n",
       "1   0/0:16:16,0:48:0,48,479:..  0/0:18:18,0:50:0,108,1079:..  \n",
       "2   0/0:16:16,0:48:0,48,479:..  0/0:18:18,0:50:0,108,1079:..  \n",
       "3   0/0:16:16,0:48:0,48,479:..    0/0:14:14,0:42:0,42,419:..  \n",
       "4   0/0:16:16,0:48:0,48,479:..    0/0:16:16,0:48:0,81,809:..  \n",
       "5   0/0:16:16,0:48:0,54,539:..    0/0:16:16,0:48:0,60,599:..  \n",
       "6   0/0:16:16,0:48:0,54,539:..    0/0:16:16,0:48:0,60,599:..  \n",
       "7   0/0:16:16,0:48:0,54,539:..    0/0:16:16,0:48:0,60,599:..  \n",
       "8   0/0:20:20,0:50:0,60,599:..    0/0:16:16,0:48:0,48,479:..  \n",
       "9   0/0:20:20,0:50:0,69,689:..    0/0:16:16,0:48:0,48,479:..  \n",
       "10  0/0:20:20,0:50:0,69,689:..    0/0:16:16,0:48:0,48,479:..  \n",
       "11  0/0:20:20,0:50:0,69,689:..    0/0:16:16,0:48:0,48,479:..  \n",
       "12  0/0:20:20,0:50:0,69,689:..    0/0:16:16,0:48:0,48,479:..  \n",
       "13  0/0:20:20,0:50:0,69,689:..    0/0:16:16,0:48:0,48,479:..  \n",
       "14  0/0:16:16,0:48:0,63,629:..    0/0:22:22,0:50:0,66,659:..  \n",
       "15  0/0:16:16,0:48:0,63,629:..    0/0:22:22,0:50:0,66,659:..  \n",
       "16  0/0:16:16,0:48:0,63,629:..    0/0:22:22,0:50:0,66,659:..  \n",
       "17  0/0:16:16,0:48:0,63,629:..    0/0:22:22,0:50:0,66,659:..  \n",
       "18  0/0:16:16,0:48:0,63,629:..    0/0:22:22,0:50:0,66,659:..  \n",
       "19  0/0:16:16,0:48:0,63,629:..    0/0:18:18,0:24:0,24,479:..  \n",
       "20  0/0:16:16,0:48:0,63,629:..    0/0:18:18,0:24:0,24,479:..  \n",
       "21  0/0:16:16,0:48:0,48,479:..    0/0:16:16,0:48:0,48,479:..  \n",
       "22  0/0:19:19,0:50:0,60,599:..    0/0:33:33,0:50:0,99,989:..  \n",
       "23  0/0:19:19,0:50:0,60,599:..    0/0:33:33,0:50:0,99,989:..  \n",
       "24  0/0:19:19,0:50:0,60,599:..    0/0:33:33,0:50:0,99,989:..  \n",
       "25  0/0:19:19,0:50:0,60,599:..    0/0:33:33,0:50:0,99,989:..  \n",
       "26  0/0:19:19,0:50:0,60,599:..    0/0:33:33,0:50:0,99,989:..  \n",
       "27  0/0:19:19,0:50:0,60,599:..    0/0:33:33,0:50:0,99,989:..  \n",
       "28  0/0:19:19,0:50:0,60,599:..    0/0:33:33,0:50:0,99,989:..  \n",
       "29  0/0:16:16,0:48:0,60,599:..    0/0:17:17,0:50:0,84,839:..  \n",
       "30  0/0:16:16,0:48:0,60,599:..    0/0:17:17,0:50:0,84,839:..  \n",
       "31  0/0:16:16,0:48:0,60,599:..    0/0:17:17,0:50:0,84,839:..  \n",
       "32  0/0:16:16,0:48:0,60,599:..    0/0:17:17,0:50:0,84,839:..  \n",
       "33  0/0:16:16,0:48:0,60,599:..    0/0:17:17,0:50:0,84,839:..  \n",
       "34  0/0:16:16,0:48:0,60,599:..    0/0:17:17,0:50:0,84,839:..  \n",
       "35  0/0:16:16,0:48:0,66,659:..    0/0:20:20,0:50:0,66,659:..  \n",
       "36  0/0:16:16,0:48:0,66,659:..    0/0:20:20,0:50:0,66,659:..  \n",
       "37                         NaN                           NaN  \n",
       "38                         NaN                           NaN  \n",
       "39                         NaN                           NaN  \n",
       "40                         NaN                           NaN  \n",
       "41                         NaN                           NaN  \n",
       "42                         NaN                           NaN  \n",
       "43                         NaN                           NaN  \n",
       "44                         NaN                           NaN  \n",
       "\n",
       "[45 rows x 41 columns]"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#This code determines the frequency of pathogenic or LOF PRKN mutations in from SNPs in WES and CNVs from microarray\n",
    "\n",
    "import pandas as pd\n",
    "import time\n",
    "import os\n",
    "import numpy as np\n",
    "import scipy.stats as stats\n",
    "\n",
    "\n",
    "#Read in SNP csv file\n",
    "SNP = pd.read_csv('/path/to/PRKN_variants_WES_touse.csv', na_values='NaN', header=0)\n",
    "SNP.drop(columns = \"number_of_var\", inplace=True)\n",
    "SNP['WESTotal'] = SNP.drop('IID', axis=1).sum(axis=1)\n",
    "\n",
    "#Read in DEL1 csv file\n",
    "DEL1 = pd.read_csv('/path/to/del_list_1only.csv', na_values='NaN', header=0, names = ['IID','B', 'C', 'DEL1', 'DEL1_Exon', 'F'])\n",
    "DEL1.drop(columns=['B','C','F'], axis=1, inplace=True)\n",
    "\n",
    "#Read in DUP1 csv file\n",
    "DUP1 = pd.read_csv('/path/todup_list_1only.csv', na_values='NaN', header=0, names = ['IID','B', 'C', 'DUP1', 'DUP1_Exon', 'F'])\n",
    "DUP1.drop(columns=['B','C','F'], axis=1, inplace=True)\n",
    "\n",
    "#Read in DELDUP2 csv file\n",
    "DELDUP2 = pd.read_csv('/path/to/2DELDUP.csv', na_values='NaN', header=0)\n",
    "\n",
    "#Merge SNP and DEL\n",
    "df = pd.merge(SNP, DEL1, on ='IID', how ='left') \n",
    "df = pd.merge(df, DUP1, on ='IID', how ='left')\n",
    "df = pd.merge(df, DELDUP2, on ='IID', how ='left') \n",
    "\n",
    "#Fix na in in DEL1 and DUP1 columns\n",
    "df['DEL1'] = df['DEL1'].fillna(0)\n",
    "df['DEL1_Exon'] = df['DEL1_Exon'].fillna('')\n",
    "df['DUP1'] = df['DUP1'].fillna(0)\n",
    "df['DUP1_Exon'] = df['DUP1_Exon'].fillna('')\n",
    "df['2DELDUP'] = df['2DELDUP'].fillna(0)\n",
    "\n",
    "#Read in key to annotate\n",
    "WESkey = pd.read_csv('/path/to/WESKey.csv', na_values='NaN', header=0)\n",
    "\n",
    "df['CNVTotal'] = df['DEL1'] + df['DUP1'] + df['2DELDUP']\n",
    "df['MutTotal'] = df['CNVTotal'] + df['WESTotal']\n",
    "\n",
    "df\n",
    "\n",
    "dfcol = df.columns\n",
    "dfcol = dfcol[1:]\n",
    "\n",
    "COLUMN_NAMES = ['ID','AA_N','AA_f','AB_N','AB_f','BB_N','BB_f','ABorBB_N','ABorBB_f','Allelic_Frequency']\n",
    "results = pd.DataFrame(columns=COLUMN_NAMES)\n",
    "\n",
    "for x in dfcol:\n",
    "\n",
    "    test = df[x].value_counts()\n",
    "    \n",
    "    testlist = []\n",
    "    \n",
    "    testlist.append(x)\n",
    "\n",
    "    try:\n",
    "        a1 = test[0]\n",
    "        testlist.append(a1)\n",
    "        \n",
    "    except:\n",
    "        a1 = 0\n",
    "        testlist.append(a1)\n",
    "    \n",
    "    try:\n",
    "        a = test[0]/test.sum()\n",
    "        testlist.append(a)\n",
    "        \n",
    "    except:\n",
    "        a = 0\n",
    "        testlist.append(a)\n",
    "\n",
    "    try:\n",
    "        b1 =  test[1]\n",
    "        testlist.append(b1)\n",
    "            \n",
    "    except:\n",
    "        b1 =  0\n",
    "        testlist.append(b1)\n",
    "        \n",
    "    try:\n",
    "        b =  test[1]/test.sum()\n",
    "        testlist.append(b)\n",
    "            \n",
    "    except:\n",
    "        b =  0\n",
    "        testlist.append(b)\n",
    "\n",
    "    try:\n",
    "        c1 = test[2] \n",
    "        testlist.append(c1)\n",
    "            \n",
    "    except:\n",
    "        c1 = 0\n",
    "        testlist.append(c1)\n",
    "                \n",
    "    try:\n",
    "        c = test[2]/test.sum() \n",
    "        testlist.append(c)\n",
    "            \n",
    "    except:\n",
    "        c = 0\n",
    "        testlist.append(c)\n",
    "\n",
    "    try:\n",
    "         testlist.append((b1+c1))\n",
    "            \n",
    "    except:\n",
    "         testlist.append(0)\n",
    "\n",
    "    try:\n",
    "         testlist.append((b+c))\n",
    "            \n",
    "    except:\n",
    "         testlist.append(0)\n",
    "    \n",
    "    try:\n",
    "         testlist.append(((b/2)+c))\n",
    "            \n",
    "    except:\n",
    "         testlist.append(0) \n",
    "    \n",
    "    results_length = len(results)\n",
    "    \n",
    "    try:\n",
    "        results.loc[results_length] = testlist\n",
    "    except:\n",
    "        pass\n",
    "    \n",
    "#Annotate the results table\n",
    "results = pd.merge(results, WESkey, on ='ID', how ='left') \n",
    "\n",
    "#Write results to file\n",
    "results.to_csv('/path/to/WESMutation_Frequency_Table.csv')\n",
    "results\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of SNPs misgenotyped as positive based on discordance with WES: 22\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>AA_N</th>\n",
       "      <th>AA_f</th>\n",
       "      <th>AB_N</th>\n",
       "      <th>AB_f</th>\n",
       "      <th>BB_N</th>\n",
       "      <th>BB_f</th>\n",
       "      <th>ABorBB_N</th>\n",
       "      <th>ABorBB_f</th>\n",
       "      <th>Allelic_Frequency</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "      <th>Otherinfo14</th>\n",
       "      <th>Otherinfo15</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6:161350101:A:G_G</td>\n",
       "      <td>190038</td>\n",
       "      <td>0.999979</td>\n",
       "      <td>4</td>\n",
       "      <td>0.000021</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0.000021</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161350101_A_G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>49.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=1.2e-05;AQ=49</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:17:17,0:50:0,51,509:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:26:26,0:50:0,78,779:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr6:161350163:C:T_T</td>\n",
       "      <td>190040</td>\n",
       "      <td>0.999989</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161350163_C_T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>45.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=5e-06;AQ=45</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:17:17,0:50:0,51,509:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:18:18,0:50:0,108,1079:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr6:161350208:C:T_T</td>\n",
       "      <td>189929</td>\n",
       "      <td>0.999405</td>\n",
       "      <td>113</td>\n",
       "      <td>0.000595</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>113</td>\n",
       "      <td>0.000595</td>\n",
       "      <td>0.000297</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161350208_C_T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>47.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=0.000314;AQ=47</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:17:17,0:50:0,51,509:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:18:18,0:50:0,108,1079:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr6:161360089:AT:A_A</td>\n",
       "      <td>190036</td>\n",
       "      <td>0.999968</td>\n",
       "      <td>6</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161360089_AT_A</td>\n",
       "      <td>AT</td>\n",
       "      <td>A</td>\n",
       "      <td>38.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=5e-06;AQ=38</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>./.:14:14,0:0:0,42,419:II</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:14:14,0:42:0,42,419:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr6:161360132:GT:G_G</td>\n",
       "      <td>190040</td>\n",
       "      <td>0.999989</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161360132_GT_G</td>\n",
       "      <td>GT</td>\n",
       "      <td>G</td>\n",
       "      <td>49.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=5e-06;AQ=49</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,57,569:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:16:16,0:48:0,81,809:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr6:161548898:G:A_A</td>\n",
       "      <td>190041</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161548898_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>43.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=43</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,87,869:..</td>\n",
       "      <td>0/0:16:16,0:48:0,54,539:..</td>\n",
       "      <td>0/0:16:16,0:48:0,60,599:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:161548965:GA:G_G</td>\n",
       "      <td>190040</td>\n",
       "      <td>0.999989</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161548965_GA_G</td>\n",
       "      <td>GA</td>\n",
       "      <td>G</td>\n",
       "      <td>39.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=5e-06;AQ=23</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,87,869:..</td>\n",
       "      <td>0/0:16:16,0:48:0,54,539:..</td>\n",
       "      <td>0/0:16:16,0:48:0,60,599:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr6:161548979:C:CACCA_CACCA</td>\n",
       "      <td>190041</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161548979_C_CACCA</td>\n",
       "      <td>C</td>\n",
       "      <td>CACCA</td>\n",
       "      <td>44.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=44</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,87,869:..</td>\n",
       "      <td>0/0:16:16,0:48:0,54,539:..</td>\n",
       "      <td>0/0:16:16,0:48:0,60,599:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr6:161569397:CA:C_C</td>\n",
       "      <td>190041</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161569397_CA_C</td>\n",
       "      <td>CA</td>\n",
       "      <td>C</td>\n",
       "      <td>40.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=37</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:18:18,0:50:0,72,719:..</td>\n",
       "      <td>0/0:20:20,0:50:0,60,599:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr6:161785793:C:G_G</td>\n",
       "      <td>190041</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161785793_C_G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>37.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=37</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:17:17,0:50:0,66,659:..</td>\n",
       "      <td>0/0:20:20,0:50:0,69,689:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr6:161785817:G:A_A</td>\n",
       "      <td>190041</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161785817_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>35.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=35</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:17:17,0:50:0,66,659:..</td>\n",
       "      <td>0/0:20:20,0:50:0,69,689:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>p.R275W</td>\n",
       "      <td>188621</td>\n",
       "      <td>0.992523</td>\n",
       "      <td>1415</td>\n",
       "      <td>0.007446</td>\n",
       "      <td>6</td>\n",
       "      <td>3.1572e-05</td>\n",
       "      <td>1421</td>\n",
       "      <td>0.007477</td>\n",
       "      <td>0.003754</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr6:161785839:A:T_T</td>\n",
       "      <td>190041</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161785839_A_T</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>44.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=44</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:20:20,0:50:0,69,689:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>p.C253Y</td>\n",
       "      <td>190040</td>\n",
       "      <td>0.999989</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>p.T240M</td>\n",
       "      <td>189969</td>\n",
       "      <td>0.999616</td>\n",
       "      <td>73</td>\n",
       "      <td>0.000384</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>73</td>\n",
       "      <td>0.000384</td>\n",
       "      <td>0.000192</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr6:161973323:C:CAAGT_CAAGT</td>\n",
       "      <td>190041</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161973323_C_CAAGT</td>\n",
       "      <td>C</td>\n",
       "      <td>CAAGT</td>\n",
       "      <td>44.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=44</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:16:16,0:48:0,63,629:..</td>\n",
       "      <td>0/0:22:22,0:50:0,66,659:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr6:161973358:CA:C_C</td>\n",
       "      <td>190041</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161973358_CA_C</td>\n",
       "      <td>CA</td>\n",
       "      <td>C</td>\n",
       "      <td>40.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=36</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:16:16,0:48:0,63,629:..</td>\n",
       "      <td>0/0:22:22,0:50:0,66,659:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr6:161973401:C:T_T</td>\n",
       "      <td>190041</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161973401_C_T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>42.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=42</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:16:16,0:48:0,63,629:..</td>\n",
       "      <td>0/0:22:22,0:50:0,66,659:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>p.K211N</td>\n",
       "      <td>190035</td>\n",
       "      <td>0.999963</td>\n",
       "      <td>7</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr6:161973414:AT:A_A</td>\n",
       "      <td>190041</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161973414_AT_A</td>\n",
       "      <td>AT</td>\n",
       "      <td>A</td>\n",
       "      <td>42.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=42</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:16:16,0:48:0,63,629:..</td>\n",
       "      <td>0/0:18:18,0:24:0,24,479:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr6:161973414:AT:ATT_ATT</td>\n",
       "      <td>190041</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_161973414_AT_ATT</td>\n",
       "      <td>AT</td>\n",
       "      <td>ATT</td>\n",
       "      <td>42.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=42</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:16:16,0:48:0,63,629:..</td>\n",
       "      <td>0/0:18:18,0:24:0,24,479:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr6:162201236:G:T_T</td>\n",
       "      <td>190041</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162201236_G_T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>45.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=45</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr6:162262537:C:CTGG_CTGG</td>\n",
       "      <td>190035</td>\n",
       "      <td>0.999963</td>\n",
       "      <td>7</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162262537_C_CTGG</td>\n",
       "      <td>C</td>\n",
       "      <td>CTGG</td>\n",
       "      <td>48.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=1.7e-05;AQ=48</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:21:21,0:48:0,66,659:..</td>\n",
       "      <td>0/0:19:19,0:50:0,60,599:..</td>\n",
       "      <td>0/0:33:33,0:50:0,99,989:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr6:162262543:G:GT_GT</td>\n",
       "      <td>190035</td>\n",
       "      <td>0.999963</td>\n",
       "      <td>7</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162262543_G_GT</td>\n",
       "      <td>G</td>\n",
       "      <td>GT</td>\n",
       "      <td>46.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=1.7e-05;AQ=46</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:21:21,0:48:0,66,659:..</td>\n",
       "      <td>0/0:19:19,0:50:0,60,599:..</td>\n",
       "      <td>0/0:33:33,0:50:0,99,989:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr6:162262560:TCAGTGTGCAGAATGACAGCCAGCCCCACAG...</td>\n",
       "      <td>189917</td>\n",
       "      <td>0.999342</td>\n",
       "      <td>125</td>\n",
       "      <td>0.000658</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>125</td>\n",
       "      <td>0.000658</td>\n",
       "      <td>0.000329</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162262560_TCAGTGTGCAGAATGACAGCCAGCCCCACAG...</td>\n",
       "      <td>TCAGTGTGCAGAATGACAGCCAGCCCCACAGAGTCTCCTGG</td>\n",
       "      <td>T</td>\n",
       "      <td>50.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=0.000202;AQ=38</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:21:21,0:48:0,66,659:..</td>\n",
       "      <td>0/0:19:19,0:50:0,60,599:..</td>\n",
       "      <td>0/0:33:33,0:50:0,99,989:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr6:162262560:TCAGTGTGCAGAATGACAGCCAGCCCCACAG...</td>\n",
       "      <td>190041</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162262560_TCAGTGTGCAGAATGACAGCCAGCCCCACAG...</td>\n",
       "      <td>TCAGTGTGCAGAATGACAGCCAGCCCCACAGAGTCTCCTGG</td>\n",
       "      <td>TCAGGTGTGCAGAATGACAGCCAGCCCCACAGAGTCTCCTGG</td>\n",
       "      <td>50.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=40</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:21:21,0:48:0,66,659:..</td>\n",
       "      <td>0/0:19:19,0:50:0,60,599:..</td>\n",
       "      <td>0/0:33:33,0:50:0,99,989:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr6:162262662:G:GAA_GAA</td>\n",
       "      <td>190040</td>\n",
       "      <td>0.999989</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162262662_G_GAA</td>\n",
       "      <td>G</td>\n",
       "      <td>GAA</td>\n",
       "      <td>39.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=5e-06;AQ=29</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:21:21,0:48:0,66,659:..</td>\n",
       "      <td>0/0:19:19,0:50:0,60,599:..</td>\n",
       "      <td>0/0:33:33,0:50:0,99,989:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr6:162262702:C:A_A</td>\n",
       "      <td>190035</td>\n",
       "      <td>0.999963</td>\n",
       "      <td>7</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162262702_C_A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>50.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=1.7e-05;AQ=50</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:21:21,0:48:0,66,659:..</td>\n",
       "      <td>0/0:19:19,0:50:0,60,599:..</td>\n",
       "      <td>0/0:33:33,0:50:0,99,989:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr6:162262715:C:CCA_CCA</td>\n",
       "      <td>190032</td>\n",
       "      <td>0.999947</td>\n",
       "      <td>10</td>\n",
       "      <td>0.000053</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0.000053</td>\n",
       "      <td>0.000026</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162262715_C_CCA</td>\n",
       "      <td>C</td>\n",
       "      <td>CCA</td>\n",
       "      <td>43.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2.5e-05;AQ=43</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:21:21,0:48:0,66,659:..</td>\n",
       "      <td>0/0:19:19,0:50:0,60,599:..</td>\n",
       "      <td>0/0:33:33,0:50:0,99,989:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr6:162443325:AT:A_A</td>\n",
       "      <td>189920</td>\n",
       "      <td>0.999358</td>\n",
       "      <td>122</td>\n",
       "      <td>0.000642</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>122</td>\n",
       "      <td>0.000642</td>\n",
       "      <td>0.000321</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162443325_AT_A</td>\n",
       "      <td>AT</td>\n",
       "      <td>A</td>\n",
       "      <td>56.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=0.000314;AQ=56</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:30:30,0:50:0,90,899:..</td>\n",
       "      <td>0/0:16:16,0:48:0,60,599:..</td>\n",
       "      <td>0/0:17:17,0:50:0,84,839:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr6:162443356:C:G_G</td>\n",
       "      <td>190027</td>\n",
       "      <td>0.999921</td>\n",
       "      <td>15</td>\n",
       "      <td>0.000079</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>0.000079</td>\n",
       "      <td>0.000039</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162443356_C_G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>49.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=4.5e-05;AQ=43</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:34:34,0:48:0,123,1229:..</td>\n",
       "      <td>0/0:16:16,0:48:0,60,599:..</td>\n",
       "      <td>0/0:17:17,0:50:0,84,839:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr6:162443378:CCT:C_C</td>\n",
       "      <td>189896</td>\n",
       "      <td>0.999232</td>\n",
       "      <td>146</td>\n",
       "      <td>0.000768</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>146</td>\n",
       "      <td>0.000768</td>\n",
       "      <td>0.000384</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162443378_CCT_C</td>\n",
       "      <td>CCT</td>\n",
       "      <td>C</td>\n",
       "      <td>55.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=0.000409;AQ=55</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:34:34,0:48:0,123,1229:..</td>\n",
       "      <td>0/0:16:16,0:48:0,60,599:..</td>\n",
       "      <td>0/0:17:17,0:50:0,84,839:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr6:162443378:CCT:CCC_CCC</td>\n",
       "      <td>189814</td>\n",
       "      <td>0.998800</td>\n",
       "      <td>228</td>\n",
       "      <td>0.001200</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>228</td>\n",
       "      <td>0.001200</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162443378_CCT_CCC</td>\n",
       "      <td>CCT</td>\n",
       "      <td>CCC</td>\n",
       "      <td>55.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=0.000635;AQ=53</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:34:34,0:48:0,123,1229:..</td>\n",
       "      <td>0/0:16:16,0:48:0,60,599:..</td>\n",
       "      <td>0/0:17:17,0:50:0,84,839:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr6:162443383:C:T_T</td>\n",
       "      <td>189951</td>\n",
       "      <td>0.999521</td>\n",
       "      <td>91</td>\n",
       "      <td>0.000479</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>0.000479</td>\n",
       "      <td>0.000239</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162443383_C_T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>50.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=0.000239;AQ=50</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:34:34,0:48:0,123,1229:..</td>\n",
       "      <td>0/0:16:16,0:48:0,60,599:..</td>\n",
       "      <td>0/0:17:17,0:50:0,84,839:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr6:162443384:G:A_A</td>\n",
       "      <td>190041</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162443384_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>35.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=35</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:34:34,0:48:0,123,1229:..</td>\n",
       "      <td>0/0:16:16,0:48:0,60,599:..</td>\n",
       "      <td>0/0:17:17,0:50:0,84,839:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr6:162727667:A:G_G</td>\n",
       "      <td>190038</td>\n",
       "      <td>0.999979</td>\n",
       "      <td>4</td>\n",
       "      <td>0.000021</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0.000021</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162727667_A_G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>42.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=1e-05;AQ=42</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:20:20,0:30:0,30,539:..</td>\n",
       "      <td>0/0:16:16,0:48:0,66,659:..</td>\n",
       "      <td>0/0:20:20,0:50:0,66,659:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr6:162727668:T:C_C</td>\n",
       "      <td>190042</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>chr6_162727668_T_C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>45.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=45</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:17:17,0:50:0,63,629:..</td>\n",
       "      <td>0/0:16:16,0:48:0,66,659:..</td>\n",
       "      <td>0/0:20:20,0:50:0,66,659:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>WESTotal</td>\n",
       "      <td>187637</td>\n",
       "      <td>0.987345</td>\n",
       "      <td>2394</td>\n",
       "      <td>0.012597</td>\n",
       "      <td>11</td>\n",
       "      <td>5.78819e-05</td>\n",
       "      <td>2405</td>\n",
       "      <td>0.012655</td>\n",
       "      <td>0.006356</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>DEL1</td>\n",
       "      <td>189481</td>\n",
       "      <td>0.997048</td>\n",
       "      <td>561</td>\n",
       "      <td>0.002952</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>561</td>\n",
       "      <td>0.002952</td>\n",
       "      <td>0.001476</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>DEL1_Exon</td>\n",
       "      <td>189481</td>\n",
       "      <td>0.997048</td>\n",
       "      <td>230</td>\n",
       "      <td>0.001210</td>\n",
       "      <td>90</td>\n",
       "      <td>0.00047358</td>\n",
       "      <td>320</td>\n",
       "      <td>0.001684</td>\n",
       "      <td>0.001079</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>MicroR275W</td>\n",
       "      <td>188599</td>\n",
       "      <td>0.992407</td>\n",
       "      <td>1437</td>\n",
       "      <td>0.007561</td>\n",
       "      <td>6</td>\n",
       "      <td>3.1572e-05</td>\n",
       "      <td>1443</td>\n",
       "      <td>0.007593</td>\n",
       "      <td>0.003812</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>MicroT240M</td>\n",
       "      <td>189969</td>\n",
       "      <td>0.999616</td>\n",
       "      <td>73</td>\n",
       "      <td>0.000384</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>73</td>\n",
       "      <td>0.000384</td>\n",
       "      <td>0.000192</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>microSNPTotal</td>\n",
       "      <td>188526</td>\n",
       "      <td>0.992023</td>\n",
       "      <td>1510</td>\n",
       "      <td>0.007946</td>\n",
       "      <td>6</td>\n",
       "      <td>3.1572e-05</td>\n",
       "      <td>1516</td>\n",
       "      <td>0.007977</td>\n",
       "      <td>0.004004</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>DUP1</td>\n",
       "      <td>189564</td>\n",
       "      <td>0.997485</td>\n",
       "      <td>478</td>\n",
       "      <td>0.002515</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>478</td>\n",
       "      <td>0.002515</td>\n",
       "      <td>0.001258</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>DUP1_Exon</td>\n",
       "      <td>189564</td>\n",
       "      <td>0.997485</td>\n",
       "      <td>304</td>\n",
       "      <td>0.001600</td>\n",
       "      <td>55</td>\n",
       "      <td>0.00028941</td>\n",
       "      <td>359</td>\n",
       "      <td>0.001889</td>\n",
       "      <td>0.001089</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>2DELDUP</td>\n",
       "      <td>190040</td>\n",
       "      <td>0.999989</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0524e-05</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>CNVTotal</td>\n",
       "      <td>189001</td>\n",
       "      <td>0.994522</td>\n",
       "      <td>1039</td>\n",
       "      <td>0.005467</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0524e-05</td>\n",
       "      <td>1041</td>\n",
       "      <td>0.005478</td>\n",
       "      <td>0.002744</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>MutTotal</td>\n",
       "      <td>186603</td>\n",
       "      <td>0.981904</td>\n",
       "      <td>3419</td>\n",
       "      <td>0.017991</td>\n",
       "      <td>20</td>\n",
       "      <td>0.00010524</td>\n",
       "      <td>3439</td>\n",
       "      <td>0.018096</td>\n",
       "      <td>0.009101</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>48 rows Ã— 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   ID    AA_N      AA_f  AB_N  \\\n",
       "0                                chr6:161350101:A:G_G  190038  0.999979     4   \n",
       "1                                chr6:161350163:C:T_T  190040  0.999989     2   \n",
       "2                                chr6:161350208:C:T_T  189929  0.999405   113   \n",
       "3                               chr6:161360089:AT:A_A  190036  0.999968     6   \n",
       "4                               chr6:161360132:GT:G_G  190040  0.999989     2   \n",
       "5                                chr6:161548898:G:A_A  190041  0.999995     1   \n",
       "6                               chr6:161548965:GA:G_G  190040  0.999989     2   \n",
       "7                        chr6:161548979:C:CACCA_CACCA  190041  0.999995     1   \n",
       "8                               chr6:161569397:CA:C_C  190041  0.999995     1   \n",
       "9                                chr6:161785793:C:G_G  190041  0.999995     1   \n",
       "10                               chr6:161785817:G:A_A  190041  0.999995     1   \n",
       "11                                            p.R275W  188621  0.992523  1415   \n",
       "12                               chr6:161785839:A:T_T  190041  0.999995     1   \n",
       "13                                            p.C253Y  190040  0.999989     2   \n",
       "14                                            p.T240M  189969  0.999616    73   \n",
       "15                       chr6:161973323:C:CAAGT_CAAGT  190041  0.999995     1   \n",
       "16                              chr6:161973358:CA:C_C  190041  0.999995     1   \n",
       "17                               chr6:161973401:C:T_T  190041  0.999995     1   \n",
       "18                                            p.K211N  190035  0.999963     7   \n",
       "19                              chr6:161973414:AT:A_A  190041  0.999995     1   \n",
       "20                          chr6:161973414:AT:ATT_ATT  190041  0.999995     1   \n",
       "21                               chr6:162201236:G:T_T  190041  0.999995     1   \n",
       "22                         chr6:162262537:C:CTGG_CTGG  190035  0.999963     7   \n",
       "23                             chr6:162262543:G:GT_GT  190035  0.999963     7   \n",
       "24  chr6:162262560:TCAGTGTGCAGAATGACAGCCAGCCCCACAG...  189917  0.999342   125   \n",
       "25  chr6:162262560:TCAGTGTGCAGAATGACAGCCAGCCCCACAG...  190041  0.999995     1   \n",
       "26                           chr6:162262662:G:GAA_GAA  190040  0.999989     2   \n",
       "27                               chr6:162262702:C:A_A  190035  0.999963     7   \n",
       "28                           chr6:162262715:C:CCA_CCA  190032  0.999947    10   \n",
       "29                              chr6:162443325:AT:A_A  189920  0.999358   122   \n",
       "30                               chr6:162443356:C:G_G  190027  0.999921    15   \n",
       "31                             chr6:162443378:CCT:C_C  189896  0.999232   146   \n",
       "32                         chr6:162443378:CCT:CCC_CCC  189814  0.998800   228   \n",
       "33                               chr6:162443383:C:T_T  189951  0.999521    91   \n",
       "34                               chr6:162443384:G:A_A  190041  0.999995     1   \n",
       "35                               chr6:162727667:A:G_G  190038  0.999979     4   \n",
       "36                               chr6:162727668:T:C_C  190042  1.000000     0   \n",
       "37                                           WESTotal  187637  0.987345  2394   \n",
       "38                                               DEL1  189481  0.997048   561   \n",
       "39                                          DEL1_Exon  189481  0.997048   230   \n",
       "40                                         MicroR275W  188599  0.992407  1437   \n",
       "41                                         MicroT240M  189969  0.999616    73   \n",
       "42                                      microSNPTotal  188526  0.992023  1510   \n",
       "43                                               DUP1  189564  0.997485   478   \n",
       "44                                          DUP1_Exon  189564  0.997485   304   \n",
       "45                                            2DELDUP  190040  0.999989     0   \n",
       "46                                           CNVTotal  189001  0.994522  1039   \n",
       "47                                           MutTotal  186603  0.981904  3419   \n",
       "\n",
       "        AB_f BB_N         BB_f ABorBB_N  ABorBB_f  Allelic_Frequency  ...  \\\n",
       "0   0.000021    0            0        4  0.000021           0.000011  ...   \n",
       "1   0.000011    0            0        2  0.000011           0.000005  ...   \n",
       "2   0.000595    0            0      113  0.000595           0.000297  ...   \n",
       "3   0.000032    0            0        6  0.000032           0.000016  ...   \n",
       "4   0.000011    0            0        2  0.000011           0.000005  ...   \n",
       "5   0.000005    0            0        1  0.000005           0.000003  ...   \n",
       "6   0.000011    0            0        2  0.000011           0.000005  ...   \n",
       "7   0.000005    0            0        1  0.000005           0.000003  ...   \n",
       "8   0.000005    0            0        1  0.000005           0.000003  ...   \n",
       "9   0.000005    0            0        1  0.000005           0.000003  ...   \n",
       "10  0.000005    0            0        1  0.000005           0.000003  ...   \n",
       "11  0.007446    6   3.1572e-05     1421  0.007477           0.003754  ...   \n",
       "12  0.000005    0            0        1  0.000005           0.000003  ...   \n",
       "13  0.000011    0            0        2  0.000011           0.000005  ...   \n",
       "14  0.000384    0            0       73  0.000384           0.000192  ...   \n",
       "15  0.000005    0            0        1  0.000005           0.000003  ...   \n",
       "16  0.000005    0            0        1  0.000005           0.000003  ...   \n",
       "17  0.000005    0            0        1  0.000005           0.000003  ...   \n",
       "18  0.000037    0            0        7  0.000037           0.000018  ...   \n",
       "19  0.000005    0            0        1  0.000005           0.000003  ...   \n",
       "20  0.000005    0            0        1  0.000005           0.000003  ...   \n",
       "21  0.000005    0            0        1  0.000005           0.000003  ...   \n",
       "22  0.000037    0            0        7  0.000037           0.000018  ...   \n",
       "23  0.000037    0            0        7  0.000037           0.000018  ...   \n",
       "24  0.000658    0            0      125  0.000658           0.000329  ...   \n",
       "25  0.000005    0            0        1  0.000005           0.000003  ...   \n",
       "26  0.000011    0            0        2  0.000011           0.000005  ...   \n",
       "27  0.000037    0            0        7  0.000037           0.000018  ...   \n",
       "28  0.000053    0            0       10  0.000053           0.000026  ...   \n",
       "29  0.000642    0            0      122  0.000642           0.000321  ...   \n",
       "30  0.000079    0            0       15  0.000079           0.000039  ...   \n",
       "31  0.000768    0            0      146  0.000768           0.000384  ...   \n",
       "32  0.001200    0            0      228  0.001200           0.000600  ...   \n",
       "33  0.000479    0            0       91  0.000479           0.000239  ...   \n",
       "34  0.000005    0            0        1  0.000005           0.000003  ...   \n",
       "35  0.000021    0            0        4  0.000021           0.000011  ...   \n",
       "36  0.000000    0            0        0  0.000000           0.000000  ...   \n",
       "37  0.012597   11  5.78819e-05     2405  0.012655           0.006356  ...   \n",
       "38  0.002952    0            0      561  0.002952           0.001476  ...   \n",
       "39  0.001210   90   0.00047358      320  0.001684           0.001079  ...   \n",
       "40  0.007561    6   3.1572e-05     1443  0.007593           0.003812  ...   \n",
       "41  0.000384    0            0       73  0.000384           0.000192  ...   \n",
       "42  0.007946    6   3.1572e-05     1516  0.007977           0.004004  ...   \n",
       "43  0.002515    0            0      478  0.002515           0.001258  ...   \n",
       "44  0.001600   55   0.00028941      359  0.001889           0.001089  ...   \n",
       "45  0.000000    2   1.0524e-05        2  0.000011           0.000011  ...   \n",
       "46  0.005467    2   1.0524e-05     1041  0.005478           0.002744  ...   \n",
       "47  0.017991   20   0.00010524     3439  0.018096           0.009101  ...   \n",
       "\n",
       "                                           Otherinfo6  \\\n",
       "0                                  chr6_161350101_A_G   \n",
       "1                                  chr6_161350163_C_T   \n",
       "2                                  chr6_161350208_C_T   \n",
       "3                                 chr6_161360089_AT_A   \n",
       "4                                 chr6_161360132_GT_G   \n",
       "5                                  chr6_161548898_G_A   \n",
       "6                                 chr6_161548965_GA_G   \n",
       "7                              chr6_161548979_C_CACCA   \n",
       "8                                 chr6_161569397_CA_C   \n",
       "9                                  chr6_161785793_C_G   \n",
       "10                                 chr6_161785817_G_A   \n",
       "11                                                NaN   \n",
       "12                                 chr6_161785839_A_T   \n",
       "13                                                NaN   \n",
       "14                                                NaN   \n",
       "15                             chr6_161973323_C_CAAGT   \n",
       "16                                chr6_161973358_CA_C   \n",
       "17                                 chr6_161973401_C_T   \n",
       "18                                                NaN   \n",
       "19                                chr6_161973414_AT_A   \n",
       "20                              chr6_161973414_AT_ATT   \n",
       "21                                 chr6_162201236_G_T   \n",
       "22                              chr6_162262537_C_CTGG   \n",
       "23                                chr6_162262543_G_GT   \n",
       "24  chr6_162262560_TCAGTGTGCAGAATGACAGCCAGCCCCACAG...   \n",
       "25  chr6_162262560_TCAGTGTGCAGAATGACAGCCAGCCCCACAG...   \n",
       "26                               chr6_162262662_G_GAA   \n",
       "27                                 chr6_162262702_C_A   \n",
       "28                               chr6_162262715_C_CCA   \n",
       "29                                chr6_162443325_AT_A   \n",
       "30                                 chr6_162443356_C_G   \n",
       "31                               chr6_162443378_CCT_C   \n",
       "32                             chr6_162443378_CCT_CCC   \n",
       "33                                 chr6_162443383_C_T   \n",
       "34                                 chr6_162443384_G_A   \n",
       "35                                 chr6_162727667_A_G   \n",
       "36                                 chr6_162727668_T_C   \n",
       "37                                                NaN   \n",
       "38                                                NaN   \n",
       "39                                                NaN   \n",
       "40                                                NaN   \n",
       "41                                                NaN   \n",
       "42                                                NaN   \n",
       "43                                                NaN   \n",
       "44                                                NaN   \n",
       "45                                                NaN   \n",
       "46                                                NaN   \n",
       "47                                                NaN   \n",
       "\n",
       "                                   Otherinfo7  \\\n",
       "0                                           A   \n",
       "1                                           C   \n",
       "2                                           C   \n",
       "3                                          AT   \n",
       "4                                          GT   \n",
       "5                                           G   \n",
       "6                                          GA   \n",
       "7                                           C   \n",
       "8                                          CA   \n",
       "9                                           C   \n",
       "10                                          G   \n",
       "11                                        NaN   \n",
       "12                                          A   \n",
       "13                                        NaN   \n",
       "14                                        NaN   \n",
       "15                                          C   \n",
       "16                                         CA   \n",
       "17                                          C   \n",
       "18                                        NaN   \n",
       "19                                         AT   \n",
       "20                                         AT   \n",
       "21                                          G   \n",
       "22                                          C   \n",
       "23                                          G   \n",
       "24  TCAGTGTGCAGAATGACAGCCAGCCCCACAGAGTCTCCTGG   \n",
       "25  TCAGTGTGCAGAATGACAGCCAGCCCCACAGAGTCTCCTGG   \n",
       "26                                          G   \n",
       "27                                          C   \n",
       "28                                          C   \n",
       "29                                         AT   \n",
       "30                                          C   \n",
       "31                                        CCT   \n",
       "32                                        CCT   \n",
       "33                                          C   \n",
       "34                                          G   \n",
       "35                                          A   \n",
       "36                                          T   \n",
       "37                                        NaN   \n",
       "38                                        NaN   \n",
       "39                                        NaN   \n",
       "40                                        NaN   \n",
       "41                                        NaN   \n",
       "42                                        NaN   \n",
       "43                                        NaN   \n",
       "44                                        NaN   \n",
       "45                                        NaN   \n",
       "46                                        NaN   \n",
       "47                                        NaN   \n",
       "\n",
       "                                    Otherinfo8 Otherinfo9 Otherinfo10  \\\n",
       "0                                            G       49.0           .   \n",
       "1                                            T       45.0           .   \n",
       "2                                            T       47.0           .   \n",
       "3                                            A       38.0           .   \n",
       "4                                            G       49.0           .   \n",
       "5                                            A       43.0           .   \n",
       "6                                            G       39.0           .   \n",
       "7                                        CACCA       44.0           .   \n",
       "8                                            C       40.0           .   \n",
       "9                                            G       37.0           .   \n",
       "10                                           A       35.0           .   \n",
       "11                                         NaN        NaN         NaN   \n",
       "12                                           T       44.0           .   \n",
       "13                                         NaN        NaN         NaN   \n",
       "14                                         NaN        NaN         NaN   \n",
       "15                                       CAAGT       44.0           .   \n",
       "16                                           C       40.0           .   \n",
       "17                                           T       42.0           .   \n",
       "18                                         NaN        NaN         NaN   \n",
       "19                                           A       42.0           .   \n",
       "20                                         ATT       42.0           .   \n",
       "21                                           T       45.0           .   \n",
       "22                                        CTGG       48.0           .   \n",
       "23                                          GT       46.0           .   \n",
       "24                                           T       50.0           .   \n",
       "25  TCAGGTGTGCAGAATGACAGCCAGCCCCACAGAGTCTCCTGG       50.0           .   \n",
       "26                                         GAA       39.0           .   \n",
       "27                                           A       50.0           .   \n",
       "28                                         CCA       43.0           .   \n",
       "29                                           A       56.0           .   \n",
       "30                                           G       49.0           .   \n",
       "31                                           C       55.0           .   \n",
       "32                                         CCC       55.0           .   \n",
       "33                                           T       50.0           .   \n",
       "34                                           A       35.0           .   \n",
       "35                                           G       42.0           .   \n",
       "36                                           C       45.0           .   \n",
       "37                                         NaN        NaN         NaN   \n",
       "38                                         NaN        NaN         NaN   \n",
       "39                                         NaN        NaN         NaN   \n",
       "40                                         NaN        NaN         NaN   \n",
       "41                                         NaN        NaN         NaN   \n",
       "42                                         NaN        NaN         NaN   \n",
       "43                                         NaN        NaN         NaN   \n",
       "44                                         NaN        NaN         NaN   \n",
       "45                                         NaN        NaN         NaN   \n",
       "46                                         NaN        NaN         NaN   \n",
       "47                                         NaN        NaN         NaN   \n",
       "\n",
       "          Otherinfo11         Otherinfo12                   Otherinfo13  \\\n",
       "0    AF=1.2e-05;AQ=49  GT:DP:AD:GQ:PL:RNC    0/0:17:17,0:50:0,51,509:..   \n",
       "1      AF=5e-06;AQ=45  GT:DP:AD:GQ:PL:RNC    0/0:17:17,0:50:0,51,509:..   \n",
       "2   AF=0.000314;AQ=47  GT:DP:AD:GQ:PL:RNC    0/0:17:17,0:50:0,51,509:..   \n",
       "3      AF=5e-06;AQ=38  GT:DP:AD:GQ:PL:RNC     ./.:14:14,0:0:0,42,419:II   \n",
       "4      AF=5e-06;AQ=49  GT:DP:AD:GQ:PL:RNC    0/0:16:16,0:48:0,57,569:..   \n",
       "5      AF=2e-06;AQ=43  GT:DP:AD:GQ:PL:RNC    0/0:16:16,0:48:0,87,869:..   \n",
       "6      AF=5e-06;AQ=23  GT:DP:AD:GQ:PL:RNC    0/0:16:16,0:48:0,87,869:..   \n",
       "7      AF=2e-06;AQ=44  GT:DP:AD:GQ:PL:RNC    0/0:16:16,0:48:0,87,869:..   \n",
       "8      AF=2e-06;AQ=37  GT:DP:AD:GQ:PL:RNC    0/0:18:18,0:50:0,72,719:..   \n",
       "9      AF=2e-06;AQ=37  GT:DP:AD:GQ:PL:RNC    0/0:17:17,0:50:0,66,659:..   \n",
       "10     AF=2e-06;AQ=35  GT:DP:AD:GQ:PL:RNC    0/0:17:17,0:50:0,66,659:..   \n",
       "11                NaN                 NaN                           NaN   \n",
       "12     AF=2e-06;AQ=44  GT:DP:AD:GQ:PL:RNC    0/0:16:16,0:48:0,48,479:..   \n",
       "13                NaN                 NaN                           NaN   \n",
       "14                NaN                 NaN                           NaN   \n",
       "15     AF=2e-06;AQ=44  GT:DP:AD:GQ:PL:RNC    0/0:16:16,0:48:0,48,479:..   \n",
       "16     AF=2e-06;AQ=36  GT:DP:AD:GQ:PL:RNC    0/0:16:16,0:48:0,48,479:..   \n",
       "17     AF=2e-06;AQ=42  GT:DP:AD:GQ:PL:RNC    0/0:16:16,0:48:0,48,479:..   \n",
       "18                NaN                 NaN                           NaN   \n",
       "19     AF=2e-06;AQ=42  GT:DP:AD:GQ:PL:RNC    0/0:16:16,0:48:0,48,479:..   \n",
       "20     AF=2e-06;AQ=42  GT:DP:AD:GQ:PL:RNC    0/0:16:16,0:48:0,48,479:..   \n",
       "21     AF=2e-06;AQ=45  GT:DP:AD:GQ:PL:RNC    0/0:16:16,0:48:0,48,479:..   \n",
       "22   AF=1.7e-05;AQ=48  GT:DP:AD:GQ:PL:RNC    0/0:21:21,0:48:0,66,659:..   \n",
       "23   AF=1.7e-05;AQ=46  GT:DP:AD:GQ:PL:RNC    0/0:21:21,0:48:0,66,659:..   \n",
       "24  AF=0.000202;AQ=38  GT:DP:AD:GQ:PL:RNC    0/0:21:21,0:48:0,66,659:..   \n",
       "25     AF=2e-06;AQ=40  GT:DP:AD:GQ:PL:RNC    0/0:21:21,0:48:0,66,659:..   \n",
       "26     AF=5e-06;AQ=29  GT:DP:AD:GQ:PL:RNC    0/0:21:21,0:48:0,66,659:..   \n",
       "27   AF=1.7e-05;AQ=50  GT:DP:AD:GQ:PL:RNC    0/0:21:21,0:48:0,66,659:..   \n",
       "28   AF=2.5e-05;AQ=43  GT:DP:AD:GQ:PL:RNC    0/0:21:21,0:48:0,66,659:..   \n",
       "29  AF=0.000314;AQ=56  GT:DP:AD:GQ:PL:RNC    0/0:30:30,0:50:0,90,899:..   \n",
       "30   AF=4.5e-05;AQ=43  GT:DP:AD:GQ:PL:RNC  0/0:34:34,0:48:0,123,1229:..   \n",
       "31  AF=0.000409;AQ=55  GT:DP:AD:GQ:PL:RNC  0/0:34:34,0:48:0,123,1229:..   \n",
       "32  AF=0.000635;AQ=53  GT:DP:AD:GQ:PL:RNC  0/0:34:34,0:48:0,123,1229:..   \n",
       "33  AF=0.000239;AQ=50  GT:DP:AD:GQ:PL:RNC  0/0:34:34,0:48:0,123,1229:..   \n",
       "34     AF=2e-06;AQ=35  GT:DP:AD:GQ:PL:RNC  0/0:34:34,0:48:0,123,1229:..   \n",
       "35     AF=1e-05;AQ=42  GT:DP:AD:GQ:PL:RNC    0/0:20:20,0:30:0,30,539:..   \n",
       "36     AF=2e-06;AQ=45  GT:DP:AD:GQ:PL:RNC    0/0:17:17,0:50:0,63,629:..   \n",
       "37                NaN                 NaN                           NaN   \n",
       "38                NaN                 NaN                           NaN   \n",
       "39                NaN                 NaN                           NaN   \n",
       "40                NaN                 NaN                           NaN   \n",
       "41                NaN                 NaN                           NaN   \n",
       "42                NaN                 NaN                           NaN   \n",
       "43                NaN                 NaN                           NaN   \n",
       "44                NaN                 NaN                           NaN   \n",
       "45                NaN                 NaN                           NaN   \n",
       "46                NaN                 NaN                           NaN   \n",
       "47                NaN                 NaN                           NaN   \n",
       "\n",
       "                   Otherinfo14                   Otherinfo15  \n",
       "0   0/0:16:16,0:48:0,48,479:..    0/0:26:26,0:50:0,78,779:..  \n",
       "1   0/0:16:16,0:48:0,48,479:..  0/0:18:18,0:50:0,108,1079:..  \n",
       "2   0/0:16:16,0:48:0,48,479:..  0/0:18:18,0:50:0,108,1079:..  \n",
       "3   0/0:16:16,0:48:0,48,479:..    0/0:14:14,0:42:0,42,419:..  \n",
       "4   0/0:16:16,0:48:0,48,479:..    0/0:16:16,0:48:0,81,809:..  \n",
       "5   0/0:16:16,0:48:0,54,539:..    0/0:16:16,0:48:0,60,599:..  \n",
       "6   0/0:16:16,0:48:0,54,539:..    0/0:16:16,0:48:0,60,599:..  \n",
       "7   0/0:16:16,0:48:0,54,539:..    0/0:16:16,0:48:0,60,599:..  \n",
       "8   0/0:20:20,0:50:0,60,599:..    0/0:16:16,0:48:0,48,479:..  \n",
       "9   0/0:20:20,0:50:0,69,689:..    0/0:16:16,0:48:0,48,479:..  \n",
       "10  0/0:20:20,0:50:0,69,689:..    0/0:16:16,0:48:0,48,479:..  \n",
       "11                         NaN                           NaN  \n",
       "12  0/0:20:20,0:50:0,69,689:..    0/0:16:16,0:48:0,48,479:..  \n",
       "13                         NaN                           NaN  \n",
       "14                         NaN                           NaN  \n",
       "15  0/0:16:16,0:48:0,63,629:..    0/0:22:22,0:50:0,66,659:..  \n",
       "16  0/0:16:16,0:48:0,63,629:..    0/0:22:22,0:50:0,66,659:..  \n",
       "17  0/0:16:16,0:48:0,63,629:..    0/0:22:22,0:50:0,66,659:..  \n",
       "18                         NaN                           NaN  \n",
       "19  0/0:16:16,0:48:0,63,629:..    0/0:18:18,0:24:0,24,479:..  \n",
       "20  0/0:16:16,0:48:0,63,629:..    0/0:18:18,0:24:0,24,479:..  \n",
       "21  0/0:16:16,0:48:0,48,479:..    0/0:16:16,0:48:0,48,479:..  \n",
       "22  0/0:19:19,0:50:0,60,599:..    0/0:33:33,0:50:0,99,989:..  \n",
       "23  0/0:19:19,0:50:0,60,599:..    0/0:33:33,0:50:0,99,989:..  \n",
       "24  0/0:19:19,0:50:0,60,599:..    0/0:33:33,0:50:0,99,989:..  \n",
       "25  0/0:19:19,0:50:0,60,599:..    0/0:33:33,0:50:0,99,989:..  \n",
       "26  0/0:19:19,0:50:0,60,599:..    0/0:33:33,0:50:0,99,989:..  \n",
       "27  0/0:19:19,0:50:0,60,599:..    0/0:33:33,0:50:0,99,989:..  \n",
       "28  0/0:19:19,0:50:0,60,599:..    0/0:33:33,0:50:0,99,989:..  \n",
       "29  0/0:16:16,0:48:0,60,599:..    0/0:17:17,0:50:0,84,839:..  \n",
       "30  0/0:16:16,0:48:0,60,599:..    0/0:17:17,0:50:0,84,839:..  \n",
       "31  0/0:16:16,0:48:0,60,599:..    0/0:17:17,0:50:0,84,839:..  \n",
       "32  0/0:16:16,0:48:0,60,599:..    0/0:17:17,0:50:0,84,839:..  \n",
       "33  0/0:16:16,0:48:0,60,599:..    0/0:17:17,0:50:0,84,839:..  \n",
       "34  0/0:16:16,0:48:0,60,599:..    0/0:17:17,0:50:0,84,839:..  \n",
       "35  0/0:16:16,0:48:0,66,659:..    0/0:20:20,0:50:0,66,659:..  \n",
       "36  0/0:16:16,0:48:0,66,659:..    0/0:20:20,0:50:0,66,659:..  \n",
       "37                         NaN                           NaN  \n",
       "38                         NaN                           NaN  \n",
       "39                         NaN                           NaN  \n",
       "40                         NaN                           NaN  \n",
       "41                         NaN                           NaN  \n",
       "42                         NaN                           NaN  \n",
       "43                         NaN                           NaN  \n",
       "44                         NaN                           NaN  \n",
       "45                         NaN                           NaN  \n",
       "46                         NaN                           NaN  \n",
       "47                         NaN                           NaN  \n",
       "\n",
       "[48 rows x 41 columns]"
      ]
     },
     "execution_count": 234,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#This code reports the frequency of mutations in WES and microarray dropping samples in which R275W or T240M was not detected by microarray.\n",
    "#Also, reports the number of calls made by microarray that were not detected by WES, which are assumed to be erroneous calls using WES as gold standard.\n",
    "\n",
    "import pandas as pd\n",
    "import time\n",
    "import os\n",
    "import numpy as np\n",
    "import scipy.stats as stats\n",
    "\n",
    "\n",
    "#Read in SNP csv file\n",
    "SNP = pd.read_csv('/path/to/PRKN_variants_WES_touse.csv', na_values='NaN', header=0)\n",
    "SNP.drop(columns = \"number_of_var\", inplace=True)\n",
    "SNP.rename(columns={\"chr6:161785820:G:A_A\": \"p.R275W\", \"chr6:161973317:G:A_A\": \"p.T240M\", \"chr6:161785885:C:T_T\":\"p.C253Y\", \"chr6:161973403:T:A_A\":\"p.K211N\"}, inplace=True)\n",
    "SNP['WESTotal'] = SNP.drop('IID', axis=1).sum(axis=1)\n",
    "\n",
    "#Read in SNP2 csv file\n",
    "SNP2 = pd.read_csv('/path/to/PRKN_pathogenic.csv', na_values='NaN', header=0, names = ['FID','IID', 'PAT', 'MAT', 'SEX', 'PHENOTYPE', 'MicroR275W', 'C253Y', 'MicroT240M', 'K211N'])\n",
    "SNP2.drop(columns=['FID','PAT','MAT','PHENOTYPE','C253Y','K211N','SEX'], axis=1, inplace=True)\n",
    "SNP2['microSNPTotal'] = SNP2.drop('IID', axis=1).sum(axis=1)\n",
    "\n",
    "#Read in DEL1 csv file\n",
    "DEL1 = pd.read_csv('/path/to/del_list_1only.csv', na_values='NaN', header=0, names = ['IID','B', 'C', 'DEL1', 'DEL1_Exon', 'F'])\n",
    "DEL1.drop(columns=['B','C','F'], axis=1, inplace=True)\n",
    "\n",
    "#Read in DUP1 csv file\n",
    "DUP1 = pd.read_csv('/path/to/dup_list_1only.csv', na_values='NaN', header=0, names = ['IID','B', 'C', 'DUP1', 'DUP1_Exon', 'F'])\n",
    "DUP1.drop(columns=['B','C','F'], axis=1, inplace=True)\n",
    "\n",
    "#Read in DELDUP2 csv file\n",
    "DELDUP2 = pd.read_csv('/path/to/New_WES_Analysis/2DELDUP.csv', na_values='NaN', header=0)\n",
    "\n",
    "#Merge SNP and DEL\n",
    "df = pd.merge(SNP, DEL1, on ='IID', how ='left') \n",
    "df = pd.merge(df, SNP2, on ='IID', how ='left') \n",
    "df = pd.merge(df, DUP1, on ='IID', how ='left')\n",
    "df = pd.merge(df, DELDUP2, on ='IID', how ='left') \n",
    "\n",
    "#Fix na in in DEL1 and DUP1 columns\n",
    "df['DEL1'] = df['DEL1'].fillna(0)\n",
    "df['DEL1_Exon'] = df['DEL1_Exon'].fillna('')\n",
    "df['DUP1'] = df['DUP1'].fillna(0)\n",
    "df['DUP1_Exon'] = df['DUP1_Exon'].fillna('')\n",
    "df['2DELDUP'] = df['2DELDUP'].fillna(0)\n",
    "\n",
    "#Read in key to annotate\n",
    "WESkey = pd.read_csv('/path/to/WESKey.csv', na_values='NaN', header=0)\n",
    "\n",
    "df['CNVTotal'] = df['DEL1'] + df['DUP1'] + df['2DELDUP']\n",
    "df['MutTotal'] = df['CNVTotal'] + df['WESTotal']\n",
    "\n",
    "df\n",
    "df.dropna(inplace=True)\n",
    "\n",
    "dfcol = df.columns\n",
    "dfcol = dfcol[1:]\n",
    "\n",
    "COLUMN_NAMES = ['ID','AA_N','AA_f','AB_N','AB_f','BB_N','BB_f','ABorBB_N','ABorBB_f','Allelic_Frequency']\n",
    "results = pd.DataFrame(columns=COLUMN_NAMES)\n",
    "\n",
    "for x in dfcol:\n",
    "\n",
    "    test = df[x].value_counts()\n",
    "    \n",
    "    testlist = []\n",
    "    \n",
    "    testlist.append(x)\n",
    "\n",
    "    try:\n",
    "        a1 = test[0]\n",
    "        testlist.append(a1)\n",
    "        \n",
    "    except:\n",
    "        a1 = 0\n",
    "        testlist.append(a1)\n",
    "    \n",
    "    try:\n",
    "        a = test[0]/test.sum()\n",
    "        testlist.append(a)\n",
    "        \n",
    "    except:\n",
    "        a = 0\n",
    "        testlist.append(a)\n",
    "\n",
    "    try:\n",
    "        b1 =  test[1]\n",
    "        testlist.append(b1)\n",
    "            \n",
    "    except:\n",
    "        b1 =  0\n",
    "        testlist.append(b1)\n",
    "        \n",
    "    try:\n",
    "        b =  test[1]/test.sum()\n",
    "        testlist.append(b)\n",
    "            \n",
    "    except:\n",
    "        b =  0\n",
    "        testlist.append(b)\n",
    "\n",
    "    try:\n",
    "        c1 = test[2] \n",
    "        testlist.append(c1)\n",
    "            \n",
    "    except:\n",
    "        c1 = 0\n",
    "        testlist.append(c1)\n",
    "                \n",
    "    try:\n",
    "        c = test[2]/test.sum() \n",
    "        testlist.append(c)\n",
    "            \n",
    "    except:\n",
    "        c = 0\n",
    "        testlist.append(c)\n",
    "\n",
    "    try:\n",
    "         testlist.append((b1+c1))\n",
    "            \n",
    "    except:\n",
    "         testlist.append(0)\n",
    "\n",
    "    try:\n",
    "         testlist.append((b+c))\n",
    "            \n",
    "    except:\n",
    "         testlist.append(0)\n",
    "    \n",
    "    try:\n",
    "         testlist.append(((b/2)+c))\n",
    "            \n",
    "    except:\n",
    "         testlist.append(0) \n",
    "    \n",
    "    results_length = len(results)\n",
    "    \n",
    "    try:\n",
    "        results.loc[results_length] = testlist\n",
    "    except:\n",
    "        pass\n",
    "    \n",
    "#Annotate the results table\n",
    "results = pd.merge(results, WESkey, on ='ID', how ='left') \n",
    "\n",
    "#Determine number of \n",
    "df\n",
    "df['FalseDet?'] = (((df['p.R275W'] != df['MicroR275W']) & (df['p.R275W'] == 0)) | ((df['p.T240M'] != df['MicroT240M']) & (df['p.T240M'] == 0)))\n",
    "FalseDet = df.loc[df['FalseDet?']==True]\n",
    "print('Number of SNPs misgenotyped as positive based on discordance with WES:', FalseDet['FalseDet?'].count())\n",
    "\n",
    "results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>AA_N</th>\n",
       "      <th>AA_f</th>\n",
       "      <th>AB_N</th>\n",
       "      <th>AB_f</th>\n",
       "      <th>BB_N</th>\n",
       "      <th>BB_f</th>\n",
       "      <th>ABorBB_N</th>\n",
       "      <th>ABorBB_f</th>\n",
       "      <th>Allelic_Frequency</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "      <th>Otherinfo14</th>\n",
       "      <th>Otherinfo15</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_161350104_C_T</td>\n",
       "      <td>200604</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>...</td>\n",
       "      <td>161350104.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>37.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=37</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:17:17,0:50:0,51,509:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:26:26,0:50:0,78,779:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr6_161350115_T_C</td>\n",
       "      <td>200628</td>\n",
       "      <td>0.999990</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>...</td>\n",
       "      <td>161350115.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>42.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=42</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:17:17,0:50:0,51,509:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:26:26,0:50:0,78,779:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr6_161350116_G_T</td>\n",
       "      <td>200628</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>...</td>\n",
       "      <td>161350116.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>43.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=43</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:17:17,0:50:0,51,509:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:26:26,0:50:0,78,779:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr6_161350119_C_G</td>\n",
       "      <td>200627</td>\n",
       "      <td>0.999990</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>...</td>\n",
       "      <td>161350119.0</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>41.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=5e-06;AQ=41</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:17:17,0:50:0,51,509:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:34:34,0:42:0,42,899:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr6_161350121_C_T</td>\n",
       "      <td>200628</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>...</td>\n",
       "      <td>161350121.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>43.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=43</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:17:17,0:50:0,51,509:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:18:18,0:50:0,108,1079:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246</th>\n",
       "      <td>chr6_162443438_C_A</td>\n",
       "      <td>200629</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>...</td>\n",
       "      <td>162443438.0</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>43.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=43</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,75,989:..</td>\n",
       "      <td>0/0:16:16,0:48:0,60,599:..</td>\n",
       "      <td>0/0:17:17,0:50:0,84,839:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>247</th>\n",
       "      <td>chr6_162443458_T_A</td>\n",
       "      <td>200624</td>\n",
       "      <td>0.999975</td>\n",
       "      <td>5</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>...</td>\n",
       "      <td>162443458.0</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>47.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=1.2e-05;AQ=47</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,75,989:..</td>\n",
       "      <td>0/0:16:16,0:48:0,60,599:..</td>\n",
       "      <td>0/0:17:17,0:50:0,84,839:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248</th>\n",
       "      <td>chr6_162727663_T_C</td>\n",
       "      <td>200621</td>\n",
       "      <td>0.999970</td>\n",
       "      <td>6</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>...</td>\n",
       "      <td>162727663.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>43.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=1.5e-05;AQ=43</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:18:18,0:50:0,54,539:..</td>\n",
       "      <td>0/0:16:16,0:48:0,66,659:..</td>\n",
       "      <td>0/0:20:20,0:50:0,66,659:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249</th>\n",
       "      <td>chr6_162727665_T_C</td>\n",
       "      <td>200624</td>\n",
       "      <td>0.999990</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>...</td>\n",
       "      <td>162727665.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>45.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=5e-06;AQ=45</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:18:18,0:50:0,54,539:..</td>\n",
       "      <td>0/0:16:16,0:48:0,66,659:..</td>\n",
       "      <td>0/0:20:20,0:50:0,66,659:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>250</th>\n",
       "      <td>WESTot</td>\n",
       "      <td>124424</td>\n",
       "      <td>0.620166</td>\n",
       "      <td>63586</td>\n",
       "      <td>0.316932</td>\n",
       "      <td>11763</td>\n",
       "      <td>0.0586303</td>\n",
       "      <td>75349</td>\n",
       "      <td>0.375562</td>\n",
       "      <td>0.217096</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>251 rows Ã— 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     ID    AA_N      AA_f   AB_N      AB_f   BB_N       BB_f  \\\n",
       "0    chr6_161350104_C_T  200604  0.999995      1  0.000005      0          0   \n",
       "1    chr6_161350115_T_C  200628  0.999990      2  0.000010      0          0   \n",
       "2    chr6_161350116_G_T  200628  0.999995      1  0.000005      0          0   \n",
       "3    chr6_161350119_C_G  200627  0.999990      2  0.000010      0          0   \n",
       "4    chr6_161350121_C_T  200628  0.999995      1  0.000005      0          0   \n",
       "..                  ...     ...       ...    ...       ...    ...        ...   \n",
       "246  chr6_162443438_C_A  200629  0.999995      1  0.000005      0          0   \n",
       "247  chr6_162443458_T_A  200624  0.999975      5  0.000025      0          0   \n",
       "248  chr6_162727663_T_C  200621  0.999970      6  0.000030      0          0   \n",
       "249  chr6_162727665_T_C  200624  0.999990      2  0.000010      0          0   \n",
       "250              WESTot  124424  0.620166  63586  0.316932  11763  0.0586303   \n",
       "\n",
       "    ABorBB_N  ABorBB_f  Allelic_Frequency  ...   Otherinfo5  Otherinfo7  \\\n",
       "0          1  0.000005           0.000002  ...  161350104.0           C   \n",
       "1          2  0.000010           0.000005  ...  161350115.0           T   \n",
       "2          1  0.000005           0.000002  ...  161350116.0           G   \n",
       "3          2  0.000010           0.000005  ...  161350119.0           C   \n",
       "4          1  0.000005           0.000002  ...  161350121.0           C   \n",
       "..       ...       ...                ...  ...          ...         ...   \n",
       "246        1  0.000005           0.000002  ...  162443438.0           C   \n",
       "247        5  0.000025           0.000012  ...  162443458.0           T   \n",
       "248        6  0.000030           0.000015  ...  162727663.0           T   \n",
       "249        2  0.000010           0.000005  ...  162727665.0           T   \n",
       "250    75349  0.375562           0.217096  ...          NaN         NaN   \n",
       "\n",
       "     Otherinfo8 Otherinfo9 Otherinfo10       Otherinfo11         Otherinfo12  \\\n",
       "0             T       37.0           .    AF=2e-06;AQ=37  GT:DP:AD:GQ:PL:RNC   \n",
       "1             C       42.0           .    AF=2e-06;AQ=42  GT:DP:AD:GQ:PL:RNC   \n",
       "2             T       43.0           .    AF=2e-06;AQ=43  GT:DP:AD:GQ:PL:RNC   \n",
       "3             G       41.0           .    AF=5e-06;AQ=41  GT:DP:AD:GQ:PL:RNC   \n",
       "4             T       43.0           .    AF=2e-06;AQ=43  GT:DP:AD:GQ:PL:RNC   \n",
       "..          ...        ...         ...               ...                 ...   \n",
       "246           A       43.0           .    AF=2e-06;AQ=43  GT:DP:AD:GQ:PL:RNC   \n",
       "247           A       47.0           .  AF=1.2e-05;AQ=47  GT:DP:AD:GQ:PL:RNC   \n",
       "248           C       43.0           .  AF=1.5e-05;AQ=43  GT:DP:AD:GQ:PL:RNC   \n",
       "249           C       45.0           .    AF=5e-06;AQ=45  GT:DP:AD:GQ:PL:RNC   \n",
       "250         NaN        NaN         NaN               NaN                 NaN   \n",
       "\n",
       "                    Otherinfo13                 Otherinfo14  \\\n",
       "0    0/0:17:17,0:50:0,51,509:..  0/0:16:16,0:48:0,48,479:..   \n",
       "1    0/0:17:17,0:50:0,51,509:..  0/0:16:16,0:48:0,48,479:..   \n",
       "2    0/0:17:17,0:50:0,51,509:..  0/0:16:16,0:48:0,48,479:..   \n",
       "3    0/0:17:17,0:50:0,51,509:..  0/0:16:16,0:48:0,48,479:..   \n",
       "4    0/0:17:17,0:50:0,51,509:..  0/0:16:16,0:48:0,48,479:..   \n",
       "..                          ...                         ...   \n",
       "246  0/0:16:16,0:48:0,75,989:..  0/0:16:16,0:48:0,60,599:..   \n",
       "247  0/0:16:16,0:48:0,75,989:..  0/0:16:16,0:48:0,60,599:..   \n",
       "248  0/0:18:18,0:50:0,54,539:..  0/0:16:16,0:48:0,66,659:..   \n",
       "249  0/0:18:18,0:50:0,54,539:..  0/0:16:16,0:48:0,66,659:..   \n",
       "250                         NaN                         NaN   \n",
       "\n",
       "                      Otherinfo15  \n",
       "0      0/0:26:26,0:50:0,78,779:..  \n",
       "1      0/0:26:26,0:50:0,78,779:..  \n",
       "2      0/0:26:26,0:50:0,78,779:..  \n",
       "3      0/0:34:34,0:42:0,42,899:..  \n",
       "4    0/0:18:18,0:50:0,108,1079:..  \n",
       "..                            ...  \n",
       "246    0/0:17:17,0:50:0,84,839:..  \n",
       "247    0/0:17:17,0:50:0,84,839:..  \n",
       "248    0/0:20:20,0:50:0,66,659:..  \n",
       "249    0/0:20:20,0:50:0,66,659:..  \n",
       "250                           NaN  \n",
       "\n",
       "[251 rows x 40 columns]"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#This code determines the frequency of all nonsyn PRKN mutations in WES \n",
    "import pandas as pd\n",
    "import time\n",
    "import os\n",
    "import numpy as np\n",
    "import scipy.stats as stats\n",
    "\n",
    "\n",
    "#Read in SNP csv file\n",
    "SNP = pd.read_csv('/path/to/PRKN_variants_AllNonsynMissense_INDI.csv', na_values='NaN', header=0)\n",
    "SNP['WESTotal'] = SNP.drop('IID', axis=1).sum(axis=1)\n",
    "\n",
    "\n",
    "#Read in key to annotate\n",
    "WESkey = pd.read_csv('/path/to/New_WES_Analysis/NewWESKey.csv', na_values='NaN', header=0)\n",
    "\n",
    "df = SNP\n",
    "\n",
    "dfcol = df.columns\n",
    "dfcol = dfcol[1:]\n",
    "\n",
    "COLUMN_NAMES = ['ID','AA_N','AA_f','AB_N','AB_f','BB_N','BB_f','ABorBB_N','ABorBB_f','Allelic_Frequency']\n",
    "results = pd.DataFrame(columns=COLUMN_NAMES)\n",
    "\n",
    "for x in dfcol:\n",
    "\n",
    "    test = df[x].value_counts()\n",
    "    \n",
    "    testlist = []\n",
    "    \n",
    "    testlist.append(x)\n",
    "\n",
    "    try:\n",
    "        a1 = test[0]\n",
    "        testlist.append(a1)\n",
    "        \n",
    "    except:\n",
    "        a1 = 0\n",
    "        testlist.append(a1)\n",
    "    \n",
    "    try:\n",
    "        a = test[0]/test.sum()\n",
    "        testlist.append(a)\n",
    "        \n",
    "    except:\n",
    "        a = 0\n",
    "        testlist.append(a)\n",
    "\n",
    "    try:\n",
    "        b1 =  test[1]\n",
    "        testlist.append(b1)\n",
    "            \n",
    "    except:\n",
    "        b1 =  0\n",
    "        testlist.append(b1)\n",
    "        \n",
    "    try:\n",
    "        b =  test[1]/test.sum()\n",
    "        testlist.append(b)\n",
    "            \n",
    "    except:\n",
    "        b =  0\n",
    "        testlist.append(b)\n",
    "\n",
    "    try:\n",
    "        c1 = test[2] \n",
    "        testlist.append(c1)\n",
    "            \n",
    "    except:\n",
    "        c1 = 0\n",
    "        testlist.append(c1)\n",
    "                \n",
    "    try:\n",
    "        c = test[2]/test.sum() \n",
    "        testlist.append(c)\n",
    "            \n",
    "    except:\n",
    "        c = 0\n",
    "        testlist.append(c)\n",
    "\n",
    "    try:\n",
    "         testlist.append((b1+c1))\n",
    "            \n",
    "    except:\n",
    "         testlist.append(0)\n",
    "\n",
    "    try:\n",
    "         testlist.append((b+c))\n",
    "            \n",
    "    except:\n",
    "         testlist.append(0)\n",
    "    \n",
    "    try:\n",
    "         testlist.append(((b/2)+c))\n",
    "            \n",
    "    except:\n",
    "         testlist.append(0) \n",
    "    \n",
    "    results_length = len(results)\n",
    "    \n",
    "    try:\n",
    "        results.loc[results_length] = testlist\n",
    "    except:\n",
    "        pass\n",
    "    \n",
    "#Annotate the results table\n",
    "results['ID'] = results['ID'].str.replace(':','_')\n",
    "results['ID'] = results['ID'].str[:-2]\n",
    "results = pd.merge(results, WESkey, on ='ID', how ='left') \n",
    "\n",
    "#Write results to file\n",
    "results.to_csv('/path/to/WESnonSynVariant_Frequency_Table.csv')\n",
    "results\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>AA_N</th>\n",
       "      <th>AA_f</th>\n",
       "      <th>AB_N</th>\n",
       "      <th>AB_f</th>\n",
       "      <th>BB_N</th>\n",
       "      <th>BB_f</th>\n",
       "      <th>ABorBB_N</th>\n",
       "      <th>ABorBB_f</th>\n",
       "      <th>Allelic_Frequency</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "      <th>Otherinfo14</th>\n",
       "      <th>Otherinfo15</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6_161350115_T_C</td>\n",
       "      <td>200628</td>\n",
       "      <td>0.999841</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>5</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>...</td>\n",
       "      <td>161350115.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>42.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=42</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:17:17,0:50:0,51,509:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:26:26,0:50:0,78,779:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr6_161350116_G_T</td>\n",
       "      <td>200628</td>\n",
       "      <td>0.999846</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>4</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>...</td>\n",
       "      <td>161350116.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>43.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=43</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:17:17,0:50:0,51,509:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:26:26,0:50:0,78,779:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr6_161350119_C_G</td>\n",
       "      <td>200627</td>\n",
       "      <td>0.999841</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>5</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>...</td>\n",
       "      <td>161350119.0</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>41.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=5e-06;AQ=41</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:17:17,0:50:0,51,509:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:34:34,0:42:0,42,899:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr6_161350121_C_T</td>\n",
       "      <td>200628</td>\n",
       "      <td>0.999846</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>4</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>...</td>\n",
       "      <td>161350121.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>43.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=43</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:17:17,0:50:0,51,509:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:18:18,0:50:0,108,1079:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr6_161350125_T_G</td>\n",
       "      <td>200572</td>\n",
       "      <td>0.999576</td>\n",
       "      <td>55</td>\n",
       "      <td>0.000274</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>58</td>\n",
       "      <td>0.000289</td>\n",
       "      <td>0.000152</td>\n",
       "      <td>...</td>\n",
       "      <td>161350125.0</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>49.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=0.000135;AQ=49</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:17:17,0:50:0,51,509:..</td>\n",
       "      <td>0/0:16:16,0:48:0,48,479:..</td>\n",
       "      <td>0/0:18:18,0:50:0,108,1079:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>213</th>\n",
       "      <td>chr6_162443438_C_A</td>\n",
       "      <td>200629</td>\n",
       "      <td>0.999846</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>4</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>...</td>\n",
       "      <td>162443438.0</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>43.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=43</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,75,989:..</td>\n",
       "      <td>0/0:16:16,0:48:0,60,599:..</td>\n",
       "      <td>0/0:17:17,0:50:0,84,839:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>214</th>\n",
       "      <td>chr6_162443458_T_A</td>\n",
       "      <td>200624</td>\n",
       "      <td>0.999826</td>\n",
       "      <td>5</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>5</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>10</td>\n",
       "      <td>0.000050</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>...</td>\n",
       "      <td>162443458.0</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>47.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=1.2e-05;AQ=47</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:16:16,0:48:0,75,989:..</td>\n",
       "      <td>0/0:16:16,0:48:0,60,599:..</td>\n",
       "      <td>0/0:17:17,0:50:0,84,839:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>215</th>\n",
       "      <td>chr6_162727663_T_C</td>\n",
       "      <td>200621</td>\n",
       "      <td>0.999821</td>\n",
       "      <td>6</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>6</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>12</td>\n",
       "      <td>0.000060</td>\n",
       "      <td>0.000045</td>\n",
       "      <td>...</td>\n",
       "      <td>162727663.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>43.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=1.5e-05;AQ=43</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:18:18,0:50:0,54,539:..</td>\n",
       "      <td>0/0:16:16,0:48:0,66,659:..</td>\n",
       "      <td>0/0:20:20,0:50:0,66,659:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>216</th>\n",
       "      <td>chr6_162727665_T_C</td>\n",
       "      <td>200624</td>\n",
       "      <td>0.999841</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>4</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>...</td>\n",
       "      <td>162727665.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>45.0</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=5e-06;AQ=45</td>\n",
       "      <td>GT:DP:AD:GQ:PL:RNC</td>\n",
       "      <td>0/0:18:18,0:50:0,54,539:..</td>\n",
       "      <td>0/0:16:16,0:48:0,66,659:..</td>\n",
       "      <td>0/0:20:20,0:50:0,66,659:..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>217</th>\n",
       "      <td>WESTotal</td>\n",
       "      <td>198642</td>\n",
       "      <td>0.989943</td>\n",
       "      <td>1978</td>\n",
       "      <td>0.009857</td>\n",
       "      <td>10</td>\n",
       "      <td>0.000050</td>\n",
       "      <td>1988</td>\n",
       "      <td>0.009907</td>\n",
       "      <td>0.004979</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>218 rows Ã— 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     ID    AA_N      AA_f  AB_N      AB_f BB_N      BB_f  \\\n",
       "0    chr6_161350115_T_C  200628  0.999841     2  0.000010    3  0.000015   \n",
       "1    chr6_161350116_G_T  200628  0.999846     1  0.000005    3  0.000015   \n",
       "2    chr6_161350119_C_G  200627  0.999841     2  0.000010    3  0.000015   \n",
       "3    chr6_161350121_C_T  200628  0.999846     1  0.000005    3  0.000015   \n",
       "4    chr6_161350125_T_G  200572  0.999576    55  0.000274    3  0.000015   \n",
       "..                  ...     ...       ...   ...       ...  ...       ...   \n",
       "213  chr6_162443438_C_A  200629  0.999846     1  0.000005    3  0.000015   \n",
       "214  chr6_162443458_T_A  200624  0.999826     5  0.000025    5  0.000025   \n",
       "215  chr6_162727663_T_C  200621  0.999821     6  0.000030    6  0.000030   \n",
       "216  chr6_162727665_T_C  200624  0.999841     2  0.000010    2  0.000010   \n",
       "217            WESTotal  198642  0.989943  1978  0.009857   10  0.000050   \n",
       "\n",
       "    ABorBB_N  ABorBB_f  Allelic_Frequency  ...   Otherinfo5  Otherinfo7  \\\n",
       "0          5  0.000025           0.000020  ...  161350115.0           T   \n",
       "1          4  0.000020           0.000017  ...  161350116.0           G   \n",
       "2          5  0.000025           0.000020  ...  161350119.0           C   \n",
       "3          4  0.000020           0.000017  ...  161350121.0           C   \n",
       "4         58  0.000289           0.000152  ...  161350125.0           T   \n",
       "..       ...       ...                ...  ...          ...         ...   \n",
       "213        4  0.000020           0.000017  ...  162443438.0           C   \n",
       "214       10  0.000050           0.000037  ...  162443458.0           T   \n",
       "215       12  0.000060           0.000045  ...  162727663.0           T   \n",
       "216        4  0.000020           0.000015  ...  162727665.0           T   \n",
       "217     1988  0.009907           0.004979  ...          NaN         NaN   \n",
       "\n",
       "     Otherinfo8 Otherinfo9 Otherinfo10        Otherinfo11         Otherinfo12  \\\n",
       "0             C       42.0           .     AF=2e-06;AQ=42  GT:DP:AD:GQ:PL:RNC   \n",
       "1             T       43.0           .     AF=2e-06;AQ=43  GT:DP:AD:GQ:PL:RNC   \n",
       "2             G       41.0           .     AF=5e-06;AQ=41  GT:DP:AD:GQ:PL:RNC   \n",
       "3             T       43.0           .     AF=2e-06;AQ=43  GT:DP:AD:GQ:PL:RNC   \n",
       "4             G       49.0           .  AF=0.000135;AQ=49  GT:DP:AD:GQ:PL:RNC   \n",
       "..          ...        ...         ...                ...                 ...   \n",
       "213           A       43.0           .     AF=2e-06;AQ=43  GT:DP:AD:GQ:PL:RNC   \n",
       "214           A       47.0           .   AF=1.2e-05;AQ=47  GT:DP:AD:GQ:PL:RNC   \n",
       "215           C       43.0           .   AF=1.5e-05;AQ=43  GT:DP:AD:GQ:PL:RNC   \n",
       "216           C       45.0           .     AF=5e-06;AQ=45  GT:DP:AD:GQ:PL:RNC   \n",
       "217         NaN        NaN         NaN                NaN                 NaN   \n",
       "\n",
       "                    Otherinfo13                 Otherinfo14  \\\n",
       "0    0/0:17:17,0:50:0,51,509:..  0/0:16:16,0:48:0,48,479:..   \n",
       "1    0/0:17:17,0:50:0,51,509:..  0/0:16:16,0:48:0,48,479:..   \n",
       "2    0/0:17:17,0:50:0,51,509:..  0/0:16:16,0:48:0,48,479:..   \n",
       "3    0/0:17:17,0:50:0,51,509:..  0/0:16:16,0:48:0,48,479:..   \n",
       "4    0/0:17:17,0:50:0,51,509:..  0/0:16:16,0:48:0,48,479:..   \n",
       "..                          ...                         ...   \n",
       "213  0/0:16:16,0:48:0,75,989:..  0/0:16:16,0:48:0,60,599:..   \n",
       "214  0/0:16:16,0:48:0,75,989:..  0/0:16:16,0:48:0,60,599:..   \n",
       "215  0/0:18:18,0:50:0,54,539:..  0/0:16:16,0:48:0,66,659:..   \n",
       "216  0/0:18:18,0:50:0,54,539:..  0/0:16:16,0:48:0,66,659:..   \n",
       "217                         NaN                         NaN   \n",
       "\n",
       "                      Otherinfo15  \n",
       "0      0/0:26:26,0:50:0,78,779:..  \n",
       "1      0/0:26:26,0:50:0,78,779:..  \n",
       "2      0/0:34:34,0:42:0,42,899:..  \n",
       "3    0/0:18:18,0:50:0,108,1079:..  \n",
       "4    0/0:18:18,0:50:0,108,1079:..  \n",
       "..                            ...  \n",
       "213    0/0:17:17,0:50:0,84,839:..  \n",
       "214    0/0:17:17,0:50:0,84,839:..  \n",
       "215    0/0:20:20,0:50:0,66,659:..  \n",
       "216    0/0:20:20,0:50:0,66,659:..  \n",
       "217                           NaN  \n",
       "\n",
       "[218 rows x 40 columns]"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#This code determines the frequency of nonsyn PRKN mutations (excluding those with annotation other than VUS or not annotated in ClinVar) in WES \n",
    "import pandas as pd\n",
    "import time\n",
    "import os\n",
    "import numpy as np\n",
    "import scipy.stats as stats\n",
    "\n",
    "\n",
    "#Read in SNP csv file\n",
    "SNP = pd.read_csv('/path/to/PRKN_variants_AllNonsynMissense_INDI.csv', na_values='NaN', header=0)\n",
    "\n",
    "tSNP = SNP.T\n",
    "\n",
    "tSNP.columns = tSNP.iloc[0]\n",
    "tSNP = tSNP.drop(['IID'])\n",
    "\n",
    "df = tSNP\n",
    "\n",
    "dfcol = df.columns\n",
    "\n",
    "df = df.reset_index()\n",
    "\n",
    "df = df.rename(columns={'index': 'ID'})\n",
    "\n",
    "df['ID'] = df['ID'].str.replace(':','_')\n",
    "\n",
    "df['ID'] = df['ID'].str[:-2]\n",
    "\n",
    "#Read in key to annotate\n",
    "WESkey = pd.read_csv('/path/to/NewWESKey.csv', na_values='NaN', header=0)\n",
    "df = pd.merge(df, WESkey, on ='ID', how ='left') \n",
    "\n",
    "df['CLNSIG'].value_counts()\n",
    "df = df.loc[(df['CLNSIG'] == '.') | (df['CLNSIG'] == 'Uncertain_significance')]\n",
    "\n",
    "df = df.T\n",
    "new_header = df.iloc[0] #grab the first row for the header\n",
    "df = df[1:] #take the data less the header row\n",
    "df.columns = new_header #set the header row as the df header\n",
    "\n",
    "df['WESTotal'] = df.sum(axis=1)\n",
    "\n",
    "\n",
    "dfcol = df.columns\n",
    "dfcol = dfcol[1:]\n",
    "\n",
    "COLUMN_NAMES = ['ID','AA_N','AA_f','AB_N','AB_f','BB_N','BB_f','ABorBB_N','ABorBB_f','Allelic_Frequency']\n",
    "results = pd.DataFrame(columns=COLUMN_NAMES)\n",
    "\n",
    "for x in dfcol:\n",
    "\n",
    "    test = df[x].value_counts()\n",
    "    \n",
    "    testlist = []\n",
    "    \n",
    "    testlist.append(x)\n",
    "\n",
    "    try:\n",
    "        a1 = test[0]\n",
    "        testlist.append(a1)\n",
    "        \n",
    "    except:\n",
    "        a1 = 0\n",
    "        testlist.append(a1)\n",
    "    \n",
    "    try:\n",
    "        a = test[0]/test.sum()\n",
    "        testlist.append(a)\n",
    "        \n",
    "    except:\n",
    "        a = 0\n",
    "        testlist.append(a)\n",
    "\n",
    "    try:\n",
    "        b1 =  test[1]\n",
    "        testlist.append(b1)\n",
    "            \n",
    "    except:\n",
    "        b1 =  0\n",
    "        testlist.append(b1)\n",
    "        \n",
    "    try:\n",
    "        b =  test[1]/test.sum()\n",
    "        testlist.append(b)\n",
    "            \n",
    "    except:\n",
    "        b =  0\n",
    "        testlist.append(b)\n",
    "\n",
    "    try:\n",
    "        c1 = test[2] \n",
    "        testlist.append(c1)\n",
    "            \n",
    "    except:\n",
    "        c1 = 0\n",
    "        testlist.append(c1)\n",
    "                \n",
    "    try:\n",
    "        c = test[2]/test.sum() \n",
    "        testlist.append(c)\n",
    "            \n",
    "    except:\n",
    "        c = 0\n",
    "        testlist.append(c)\n",
    "\n",
    "    try:\n",
    "         testlist.append((b1+c1))\n",
    "            \n",
    "    except:\n",
    "         testlist.append(0)\n",
    "\n",
    "    try:\n",
    "         testlist.append((b+c))\n",
    "            \n",
    "    except:\n",
    "         testlist.append(0)\n",
    "    \n",
    "    try:\n",
    "         testlist.append(((b/2)+c))\n",
    "            \n",
    "    except:\n",
    "         testlist.append(0) \n",
    "    \n",
    "    results_length = len(results)\n",
    "    \n",
    "    try:\n",
    "        results.loc[results_length] = testlist\n",
    "    except:\n",
    "        pass\n",
    "\n",
    "results = pd.merge(results, WESkey, on ='ID', how ='left') \n",
    "\n",
    "#Write results to file\n",
    "results.to_csv('/path/to/WESnonSynVariantVUSonly_Frequency_Table.csv')\n",
    "\n",
    "\n",
    "SNP2 = df \n",
    "\n",
    "results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Mut_Cases</th>\n",
       "      <th>Mut_CTRLs</th>\n",
       "      <th>NoMut_Cases</th>\n",
       "      <th>NoMut_CTRLs</th>\n",
       "      <th>Odds</th>\n",
       "      <th>p-value</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mutation</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>WES_SNP</th>\n",
       "      <td>15</td>\n",
       "      <td>2517</td>\n",
       "      <td>1359</td>\n",
       "      <td>196715</td>\n",
       "      <td>0.8626329919023049</td>\n",
       "      <td>0.7149317662747136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AllMut</th>\n",
       "      <td>23</td>\n",
       "      <td>3648</td>\n",
       "      <td>1351</td>\n",
       "      <td>195584</td>\n",
       "      <td>0.9127481917228304</td>\n",
       "      <td>0.7617017238515316</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Mut_Cases Mut_CTRLs NoMut_Cases NoMut_CTRLs                Odds  \\\n",
       "Mutation                                                                   \n",
       "WES_SNP         15      2517        1359      196715  0.8626329919023049   \n",
       "AllMut          23      3648        1351      195584  0.9127481917228304   \n",
       "\n",
       "                     p-value  \n",
       "Mutation                      \n",
       "WES_SNP   0.7149317662747136  \n",
       "AllMut    0.7617017238515316  "
      ]
     },
     "execution_count": 199,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#This code calculates the odds ratio and p-value for a single pathogenic or LOF PRKN mutations (detected by WES or microarray) in PD cases \n",
    "\n",
    "\n",
    "import pandas as pd\n",
    "import time\n",
    "import os\n",
    "import numpy as np\n",
    "import scipy.stats as stats\n",
    "\n",
    "\n",
    "#Read in SNP csv file\n",
    "SNP = pd.read_csv('/path/to/PRKN_variants_WES_touse.csv', na_values='NaN', header=0)\n",
    "SNP.drop(columns = \"number_of_var\", inplace=True)\n",
    "SNP['WESTotal'] = SNP.drop('IID', axis=1).sum(axis=1)\n",
    "\n",
    "#Read in DEL1 csv file\n",
    "DEL1 = pd.read_csv('/path/to/del_list_1only.csv', na_values='NaN', header=0, names = ['IID','B', 'C', 'DEL1', 'DEL1_Exon', 'F'])\n",
    "DEL1.drop(columns=['B','C','F'], axis=1, inplace=True)\n",
    "\n",
    "#Read in DUP1 csv file\n",
    "DUP1 = pd.read_csv('/path/to/dup_list_1only.csv', na_values='NaN', header=0, names = ['IID','B', 'C', 'DUP1', 'DUP1_Exon', 'F'])\n",
    "DUP1.drop(columns=['B','C','F'], axis=1, inplace=True)\n",
    "\n",
    "#Read in DELDUP2 csv file\n",
    "DELDUP2 = pd.read_csv('/path/to/2DELDUP.csv', na_values='NaN', header=0)\n",
    "\n",
    "#Merge SNP and DEL\n",
    "df = pd.merge(SNP, DEL1, on ='IID', how ='left') \n",
    "df = pd.merge(df, DUP1, on ='IID', how ='left')\n",
    "df = pd.merge(df, DELDUP2, on ='IID', how ='left') \n",
    "\n",
    "#Fix na in in DEL1 and DUP1 columns\n",
    "df['DEL1'] = df['DEL1'].fillna(0)\n",
    "df['DEL1_Exon'] = df['DEL1_Exon'].fillna('')\n",
    "df['DUP1'] = df['DUP1'].fillna(0)\n",
    "df['DUP1_Exon'] = df['DUP1_Exon'].fillna('')\n",
    "df['2DELDUP'] = df['2DELDUP'].fillna(0)\n",
    "\n",
    "df['CNVTotal'] = df['DEL1'] + df['DUP1'] + df['2DELDUP']\n",
    "df['MutTotal'] = df['CNVTotal'] + df['WESTotal']\n",
    "\n",
    "#Read in PDStatus csv file\n",
    "PDStatus = pd.read_csv('/path/to/PD_April2021_with_covariates.csv', na_values='NaN', header=0)\n",
    "PDStatus['PD'] = 1\n",
    "\n",
    "FPDStatus = pd.read_csv('/path/to/ALL_indi_with_PD_parent_UNIQUE.csv', na_values='NaN', header=0)\n",
    "FPDStatus['FPD'] = 1\n",
    "\n",
    "#Merge SNP and DEL\n",
    "df2 = pd.merge(df, PDStatus, on ='IID', how ='left') \n",
    "df2 = pd.merge(df2, FPDStatus, on = 'IID', how='left')\n",
    "\n",
    "df2['PD'] = df2['PD'].fillna(0)\n",
    "df2['FPD'] = df2['FPD'].fillna(0)\n",
    "\n",
    "df2['PD'].value_counts()\n",
    "\n",
    "#Limit to cases with only 0 or 1 detected PRKN path or LOF variant\n",
    "df2 = df2.loc[df2['MutTotal'] <= 1.0]\n",
    "\n",
    "#PD in WESTotal\n",
    "PDWESTotal = df2.loc[(df2['WESTotal'] == 1) & (df2['PD'] == 1)]\n",
    "PDWESTotalc = PDWESTotal['PD'].count()\n",
    "\n",
    "NoPDWESTotal = df2.loc[(df2['WESTotal'] == 1) & (df2['PD'] == 0)]\n",
    "NoPDWESTotalc = NoPDWESTotal['PD'].count()\n",
    "\n",
    "PDNoWESTotal = df2.loc[(df2['WESTotal'] == 0) & (df2['PD'] == 1)]\n",
    "PDNoWESTotalc = PDNoWESTotal['PD'].count()\n",
    "\n",
    "NoPDNoWESTotal = df2.loc[(df2['WESTotal'] == 0) & (df2['PD'] == 0)]\n",
    "NoPDNoWESTotalc = NoPDNoWESTotal['PD'].count()\n",
    "\n",
    "testWESTotal, pvalueWESTotal = stats.fisher_exact([[PDWESTotalc, NoPDWESTotalc], [PDNoWESTotalc, NoPDNoWESTotalc]])\n",
    "\n",
    "\n",
    "#PD in MutTotal\n",
    "PDMutTotal = df2.loc[(df2['MutTotal'] == 1) & (df2['PD'] == 1)]\n",
    "PDMutTotalc = PDMutTotal['PD'].count()\n",
    "\n",
    "NoPDMutTotal = df2.loc[(df2['MutTotal'] == 1) & (df2['PD'] == 0)]\n",
    "NoPDMutTotalc = NoPDMutTotal['PD'].count()\n",
    "\n",
    "PDNoMutTotal = df2.loc[(df2['MutTotal'] == 0) & (df2['PD'] == 1)]\n",
    "PDNoMutTotalc = PDNoMutTotal['PD'].count()\n",
    "\n",
    "NoPDNoMutTotal = df2.loc[(df2['MutTotal'] == 0) & (df2['PD'] == 0)]\n",
    "NoPDNoMutTotalc = NoPDNoMutTotal['PD'].count()\n",
    "\n",
    "testMutTotal, pvalueMutTotal = stats.fisher_exact([[PDMutTotalc, NoPDMutTotalc], [PDNoMutTotalc, NoPDNoMutTotalc]])\n",
    "\n",
    "\n",
    "#Make pandas dataframe with result\n",
    "results = pd.DataFrame(np.array([['WES_SNP', PDWESTotalc, NoPDWESTotalc, PDNoWESTotalc, NoPDNoWESTotalc, testWESTotal, pvalueWESTotal], ['AllMut', PDMutTotalc, NoPDMutTotalc, PDNoMutTotalc, NoPDNoMutTotalc, testMutTotal, pvalueMutTotal]]), columns=['Mutation','Mut_Cases', 'Mut_CTRLs', 'NoMut_Cases','NoMut_CTRLs','Odds','p-value'])\n",
    "results.set_index('Mutation', inplace=True)\n",
    "\n",
    "#Write results to file\n",
    "results.to_csv('/path/to/WESMutation_PD_Odds_Table.csv')\n",
    "results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Mut_Cases</th>\n",
       "      <th>Mut_CTRLs</th>\n",
       "      <th>NoMut_Cases</th>\n",
       "      <th>NoMut_CTRLs</th>\n",
       "      <th>Odds</th>\n",
       "      <th>p-value</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mutation</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>WES_SNP</th>\n",
       "      <td>104</td>\n",
       "      <td>2428</td>\n",
       "      <td>7390</td>\n",
       "      <td>190684</td>\n",
       "      <td>1.105234599496626</td>\n",
       "      <td>0.31617807711940715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AllMut</th>\n",
       "      <td>152</td>\n",
       "      <td>3519</td>\n",
       "      <td>7342</td>\n",
       "      <td>189593</td>\n",
       "      <td>1.1154041077858152</td>\n",
       "      <td>0.1876711012182919</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Mut_Cases Mut_CTRLs NoMut_Cases NoMut_CTRLs                Odds  \\\n",
       "Mutation                                                                   \n",
       "WES_SNP        104      2428        7390      190684   1.105234599496626   \n",
       "AllMut         152      3519        7342      189593  1.1154041077858152   \n",
       "\n",
       "                      p-value  \n",
       "Mutation                       \n",
       "WES_SNP   0.31617807711940715  \n",
       "AllMut     0.1876711012182919  "
      ]
     },
     "execution_count": 197,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#This code calculates the odds ratio and p-value for a single pathogenic or LOF PRKN mutations (detected by WES or microarray) in those with parent with PD\n",
    "\n",
    "import pandas as pd\n",
    "import time\n",
    "import os\n",
    "import numpy as np\n",
    "import scipy.stats as stats\n",
    "\n",
    "\n",
    "#Read in SNP csv file\n",
    "SNP = pd.read_csv('/path/to/PRKN_variants_WES_touse.csv', na_values='NaN', header=0)\n",
    "SNP.drop(columns = \"number_of_var\", inplace=True)\n",
    "SNP['WESTotal'] = SNP.drop('IID', axis=1).sum(axis=1)\n",
    "\n",
    "#Read in DEL1 csv file\n",
    "DEL1 = pd.read_csv('/path/to/del_list_1only.csv', na_values='NaN', header=0, names = ['IID','B', 'C', 'DEL1', 'DEL1_Exon', 'F'])\n",
    "DEL1.drop(columns=['B','C','F'], axis=1, inplace=True)\n",
    "\n",
    "#Read in DUP1 csv file\n",
    "DUP1 = pd.read_csv('/path/to/dup_list_1only.csv', na_values='NaN', header=0, names = ['IID','B', 'C', 'DUP1', 'DUP1_Exon', 'F'])\n",
    "DUP1.drop(columns=['B','C','F'], axis=1, inplace=True)\n",
    "\n",
    "#Read in DELDUP2 csv file\n",
    "DELDUP2 = pd.read_csv('/path/to/2DELDUP.csv', na_values='NaN', header=0)\n",
    "\n",
    "#Merge SNP and DEL\n",
    "df = pd.merge(SNP, DEL1, on ='IID', how ='left') \n",
    "df = pd.merge(df, DUP1, on ='IID', how ='left')\n",
    "df = pd.merge(df, DELDUP2, on ='IID', how ='left') \n",
    "\n",
    "#Fix na in in DEL1 and DUP1 columns\n",
    "df['DEL1'] = df['DEL1'].fillna(0)\n",
    "df['DEL1_Exon'] = df['DEL1_Exon'].fillna('')\n",
    "df['DUP1'] = df['DUP1'].fillna(0)\n",
    "df['DUP1_Exon'] = df['DUP1_Exon'].fillna('')\n",
    "df['2DELDUP'] = df['2DELDUP'].fillna(0)\n",
    "\n",
    "df['CNVTotal'] = df['DEL1'] + df['DUP1'] + df['2DELDUP']\n",
    "df['MutTotal'] = df['CNVTotal'] + df['WESTotal']\n",
    "\n",
    "#Read in PDStatus csv file\n",
    "PDStatus = pd.read_csv('/path/to/PD_April2021_with_covariates.csv', na_values='NaN', header=0)\n",
    "PDStatus['PD'] = 1\n",
    "\n",
    "FPDStatus = pd.read_csv('/path/to/ALL_indi_with_PD_parent_UNIQUE.csv', na_values='NaN', header=0)\n",
    "FPDStatus['FPD'] = 1\n",
    "\n",
    "#Merge SNP and DEL\n",
    "df2 = pd.merge(df, PDStatus, on ='IID', how ='left') \n",
    "df2 = pd.merge(df2, FPDStatus, on = 'IID', how='left')\n",
    "\n",
    "df2['PD'] = df2['PD'].fillna(0)\n",
    "df2['FPD'] = df2['FPD'].fillna(0)\n",
    "\n",
    "df2['PD'].value_counts()\n",
    "\n",
    "#Limit to cases with only 0 or 1 detected PRKN path or LOF variant\n",
    "df2 = df2.loc[df2['MutTotal'] <= 1.0]\n",
    "\n",
    "#PD in WESTotal\n",
    "PDWESTotal = df2.loc[(df2['WESTotal'] == 1) & (df2['FPD'] == 1)]\n",
    "PDWESTotalc = PDWESTotal['FPD'].count()\n",
    "\n",
    "NoPDWESTotal = df2.loc[(df2['WESTotal'] == 1) & (df2['FPD'] == 0)]\n",
    "NoPDWESTotalc = NoPDWESTotal['FPD'].count()\n",
    "\n",
    "PDNoWESTotal = df2.loc[(df2['WESTotal'] == 0) & (df2['FPD'] == 1)]\n",
    "PDNoWESTotalc = PDNoWESTotal['FPD'].count()\n",
    "\n",
    "NoPDNoWESTotal = df2.loc[(df2['WESTotal'] == 0) & (df2['FPD'] == 0)]\n",
    "NoPDNoWESTotalc = NoPDNoWESTotal['FPD'].count()\n",
    "\n",
    "testWESTotal, pvalueWESTotal = stats.fisher_exact([[PDWESTotalc, NoPDWESTotalc], [PDNoWESTotalc, NoPDNoWESTotalc]])\n",
    "\n",
    "\n",
    "#PD in MutTotal\n",
    "PDMutTotal = df2.loc[(df2['MutTotal'] == 1) & (df2['FPD'] == 1)]\n",
    "PDMutTotalc = PDMutTotal['FPD'].count()\n",
    "\n",
    "NoPDMutTotal = df2.loc[(df2['MutTotal'] == 1) & (df2['FPD'] == 0)]\n",
    "NoPDMutTotalc = NoPDMutTotal['FPD'].count()\n",
    "\n",
    "PDNoMutTotal = df2.loc[(df2['MutTotal'] == 0) & (df2['FPD'] == 1)]\n",
    "PDNoMutTotalc = PDNoMutTotal['FPD'].count()\n",
    "\n",
    "NoPDNoMutTotal = df2.loc[(df2['MutTotal'] == 0) & (df2['FPD'] == 0)]\n",
    "NoPDNoMutTotalc = NoPDNoMutTotal['FPD'].count()\n",
    "\n",
    "testMutTotal, pvalueMutTotal = stats.fisher_exact([[PDMutTotalc, NoPDMutTotalc], [PDNoMutTotalc, NoPDNoMutTotalc]])\n",
    "\n",
    "\n",
    "#Make pandas dataframe with result\n",
    "results = pd.DataFrame(np.array([['WES_SNP', PDWESTotalc, NoPDWESTotalc, PDNoWESTotalc, NoPDNoWESTotalc, testWESTotal, pvalueWESTotal], ['AllMut', PDMutTotalc, NoPDMutTotalc, PDNoMutTotalc, NoPDNoMutTotalc, testMutTotal, pvalueMutTotal]]), columns=['Mutation','Mut_Cases', 'Mut_CTRLs', 'NoMut_Cases','NoMut_CTRLs','Odds','p-value'])\n",
    "results.set_index('Mutation', inplace=True)\n",
    "\n",
    "#Write results to file\n",
    "results.to_csv('/path/to/WESMutation_FPD_Odds_Table.csv')\n",
    "results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 271,
   "metadata": {},
   "outputs": [],
   "source": [
    "#This cell writes to .csv file table with only biallelic PRKN patients\n",
    "\n",
    "results = df2.loc[(df2['MutTotal'] == 2)]\n",
    "\n",
    "#Write results to file\n",
    "results.to_csv('/path/to/WESMutation_Biallelic.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C253Y- mismatch: 43 , match: 2\n",
      "C253Y discordant rate: 0.9555555555555556\n",
      "C253Y concordant rate: 0.044444444444444446\n",
      "K211N- mismatch: 3 , match: 7\n",
      "K211N discordant rate: 0.3\n",
      "K211N concordant rate: 0.7\n",
      "R275W- mismatch: 21 , match: 1431\n",
      "R275W discordant rate: 0.014462809917355372\n",
      "R275W concordant rate: 0.9855371900826446\n",
      "T240M- mismatch: 0 , match: 74\n",
      "T240M discordant rate: 0.0\n",
      "T240M concordant rate: 1.0\n"
     ]
    }
   ],
   "source": [
    "#This code checks concordance between WES and microarray calls for R275W, T240M, C253Y, and K211N variants \n",
    "\n",
    "import pandas as pd\n",
    "import time\n",
    "import os\n",
    "import numpy as np\n",
    "import scipy.stats as stats\n",
    "\n",
    "\n",
    "#Read in SNP csv file\n",
    "WESSNP = pd.read_csv('/path/to/PRKN_variants_WES_touse.csv', na_values='NaN', header=0)\n",
    "WESSNP.drop(columns = \"number_of_var\", inplace=True)\n",
    "WESSNP.rename(columns={\"chr6:161785820:G:A_A\": \"p.R275W\", \"chr6:161973317:G:A_A\": \"p.T240M\", \"chr6:161785885:C:T_T\":\"p.C253Y\", \"chr6:161973403:T:A_A\":\"p.K211N\"}, inplace=True)\n",
    "WESSNP = WESSNP[['IID','p.R275W','p.T240M','p.C253Y','p.K211N']]\n",
    "\n",
    "#Read in SNP from microarray\n",
    "SNP = pd.read_csv('/path/to/PRKN_pathogenic.csv', na_values='NaN', header=0, names = ['FID','IID', 'PAT', 'MAT', 'SEX', 'PHENOTYPE', 'R275W', 'C253Y', 'T240M', 'K211N'])\n",
    "SNP.drop(columns=['FID','PAT','MAT','PHENOTYPE', 'SEX'], axis=1, inplace=True)\n",
    "\n",
    "df = pd.merge(WESSNP, SNP, on ='IID', how ='left') \n",
    "\n",
    "df['R275W_match?'] = df['p.R275W'] == df['R275W']\n",
    "df['T240M_match?'] = df['p.T240M'] == df['T240M']\n",
    "df['C253Y_match?'] = df['p.C253Y'] == df['C253Y']\n",
    "df['K211N_match?'] = df['p.K211N'] == df['K211N']\n",
    "\n",
    "df.dropna(inplace=True)\n",
    "\n",
    "mR275Wmismtach = df.loc[((df['R275W'] == 1) & (df['R275W_match?'] == False))]\n",
    "mR275Wmatch = df.loc[((df['R275W'] == 1) & (df['R275W_match?'] == True))]\n",
    "\n",
    "mT240Mmismtach = df.loc[((df['T240M'] == 1) & (df['T240M_match?'] == False))]\n",
    "mT240Mmatch = df.loc[((df['T240M'] == 1) & (df['T240M_match?'] == True))]\n",
    "\n",
    "mC253Ymismtach = df.loc[((df['C253Y'] == 1) & (df['C253Y_match?'] == False))]\n",
    "mC253Ymatch = df.loc[((df['C253Y'] == 1) & (df['C253Y_match?'] == True))]\n",
    "\n",
    "mK211Nmismtach = df.loc[((df['K211N'] == 1) & (df['K211N_match?'] == False))]\n",
    "mK211Nmatch = df.loc[((df['K211N'] == 1) & (df['K211N_match?'] == True))]\n",
    "\n",
    "print('C253Y-', 'mismatch:', mC253Ymismtach['C253Y_match?'].count(), ', match:', mC253Ymatch['C253Y_match?'].count())\n",
    "print('C253Y discordant rate:', mC253Ymismtach['C253Y_match?'].count()/(mC253Ymismtach['C253Y_match?'].count() + mC253Ymatch['C253Y_match?'].count()))\n",
    "print('C253Y concordant rate:', mC253Ymatch['C253Y_match?'].count()/(mC253Ymismtach['C253Y_match?'].count() + mC253Ymatch['C253Y_match?'].count()))\n",
    "\n",
    "print('K211N-', 'mismatch:', mK211Nmismtach['K211N_match?'].count(), ', match:', mK211Nmatch['K211N_match?'].count())\n",
    "print('K211N discordant rate:', mK211Nmismtach['K211N_match?'].count()/(mK211Nmismtach['K211N_match?'].count() + mK211Nmatch['K211N_match?'].count()))\n",
    "print('K211N concordant rate:', mK211Nmatch['K211N_match?'].count()/(mK211Nmismtach['K211N_match?'].count() + mK211Nmatch['K211N_match?'].count()))\n",
    "\n",
    "print('R275W-', 'mismatch:', mR275Wmismtach['R275W_match?'].count(), ', match:', mR275Wmatch['R275W_match?'].count())\n",
    "print('R275W discordant rate:', mR275Wmismtach['R275W_match?'].count()/(mR275Wmismtach['R275W_match?'].count() + mR275Wmatch['R275W_match?'].count()))\n",
    "print('R275W concordant rate:', mR275Wmatch['R275W_match?'].count()/(mR275Wmismtach['R275W_match?'].count() + mR275Wmatch['R275W_match?'].count()))\n",
    "\n",
    "print('T240M-', 'mismatch:', mT240MWmismtach['R275W_match?'].count(), ', match:', mT240Mmatch['R275W_match?'].count())\n",
    "print('T240M discordant rate:', mT240Mmismtach['T240M_match?'].count()/(mT240Mmismtach['T240M_match?'].count() + mT240Mmatch['T240M_match?'].count()))\n",
    "print('T240M concordant rate:', mT240Mmatch['T240M_match?'].count()/(mT240Mmismtach['T240M_match?'].count() + mT240Mmatch['T240M_match?'].count()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 328,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>AA_N</th>\n",
       "      <th>AA_f</th>\n",
       "      <th>AB_N</th>\n",
       "      <th>AB_f</th>\n",
       "      <th>BB_N</th>\n",
       "      <th>BB_f</th>\n",
       "      <th>ABorBB_N</th>\n",
       "      <th>ABorBB_f</th>\n",
       "      <th>Allelic_Frequency</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>R275W</td>\n",
       "      <td>479457</td>\n",
       "      <td>0.992200</td>\n",
       "      <td>3759</td>\n",
       "      <td>0.007779</td>\n",
       "      <td>10</td>\n",
       "      <td>0.000021</td>\n",
       "      <td>3769</td>\n",
       "      <td>0.007800</td>\n",
       "      <td>0.003910</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>T240M</td>\n",
       "      <td>487615</td>\n",
       "      <td>0.999565</td>\n",
       "      <td>212</td>\n",
       "      <td>0.000435</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>212</td>\n",
       "      <td>0.000435</td>\n",
       "      <td>0.000217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>WESTotal</td>\n",
       "      <td>484397</td>\n",
       "      <td>0.991851</td>\n",
       "      <td>3969</td>\n",
       "      <td>0.008127</td>\n",
       "      <td>11</td>\n",
       "      <td>0.000023</td>\n",
       "      <td>3980</td>\n",
       "      <td>0.008149</td>\n",
       "      <td>0.004086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DEL1</td>\n",
       "      <td>486929</td>\n",
       "      <td>0.997035</td>\n",
       "      <td>1448</td>\n",
       "      <td>0.002965</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1448</td>\n",
       "      <td>0.002965</td>\n",
       "      <td>0.001482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>DEL1_Exon</td>\n",
       "      <td>486929</td>\n",
       "      <td>0.997035</td>\n",
       "      <td>533</td>\n",
       "      <td>0.001091</td>\n",
       "      <td>225</td>\n",
       "      <td>0.000461</td>\n",
       "      <td>758</td>\n",
       "      <td>0.001552</td>\n",
       "      <td>0.001006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>DUP1</td>\n",
       "      <td>487144</td>\n",
       "      <td>0.997475</td>\n",
       "      <td>1233</td>\n",
       "      <td>0.002525</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1233</td>\n",
       "      <td>0.002525</td>\n",
       "      <td>0.001262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>DUP1_Exon</td>\n",
       "      <td>487144</td>\n",
       "      <td>0.997475</td>\n",
       "      <td>783</td>\n",
       "      <td>0.001603</td>\n",
       "      <td>142</td>\n",
       "      <td>0.000291</td>\n",
       "      <td>925</td>\n",
       "      <td>0.001894</td>\n",
       "      <td>0.001092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2DELDUP</td>\n",
       "      <td>488371</td>\n",
       "      <td>0.999988</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>6</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.000012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CNVTotal</td>\n",
       "      <td>485690</td>\n",
       "      <td>0.994498</td>\n",
       "      <td>2681</td>\n",
       "      <td>0.005490</td>\n",
       "      <td>6</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>2687</td>\n",
       "      <td>0.005502</td>\n",
       "      <td>0.002757</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MutTotal</td>\n",
       "      <td>481729</td>\n",
       "      <td>0.986388</td>\n",
       "      <td>6612</td>\n",
       "      <td>0.013539</td>\n",
       "      <td>36</td>\n",
       "      <td>0.000074</td>\n",
       "      <td>6648</td>\n",
       "      <td>0.013612</td>\n",
       "      <td>0.006843</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          ID    AA_N      AA_f  AB_N      AB_f BB_N      BB_f ABorBB_N  \\\n",
       "0      R275W  479457  0.992200  3759  0.007779   10  0.000021     3769   \n",
       "1      T240M  487615  0.999565   212  0.000435    0  0.000000      212   \n",
       "2   WESTotal  484397  0.991851  3969  0.008127   11  0.000023     3980   \n",
       "3       DEL1  486929  0.997035  1448  0.002965    0  0.000000     1448   \n",
       "4  DEL1_Exon  486929  0.997035   533  0.001091  225  0.000461      758   \n",
       "5       DUP1  487144  0.997475  1233  0.002525    0  0.000000     1233   \n",
       "6  DUP1_Exon  487144  0.997475   783  0.001603  142  0.000291      925   \n",
       "7    2DELDUP  488371  0.999988     0  0.000000    6  0.000012        6   \n",
       "8   CNVTotal  485690  0.994498  2681  0.005490    6  0.000012     2687   \n",
       "9   MutTotal  481729  0.986388  6612  0.013539   36  0.000074     6648   \n",
       "\n",
       "   ABorBB_f  Allelic_Frequency  \n",
       "0  0.007800           0.003910  \n",
       "1  0.000435           0.000217  \n",
       "2  0.008149           0.004086  \n",
       "3  0.002965           0.001482  \n",
       "4  0.001552           0.001006  \n",
       "5  0.002525           0.001262  \n",
       "6  0.001894           0.001092  \n",
       "7  0.000012           0.000012  \n",
       "8  0.005502           0.002757  \n",
       "9  0.013612           0.006843  "
      ]
     },
     "execution_count": 328,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#This code determines the frequency of pathogenic or LOF PRKN mutations in from SNPs in microarray and CNVs from microarray\n",
    "\n",
    "import pandas as pd\n",
    "import time\n",
    "import os\n",
    "import numpy as np\n",
    "import scipy.stats as stats\n",
    "\n",
    "\n",
    "#Read in SNP csv file\n",
    "SNP = pd.read_csv('/path/to/PRKN_pathogenic.csv', na_values='NaN', header=0, names = ['FID','IID', 'PAT', 'MAT', 'SEX', 'PHENOTYPE', 'R275W', 'C253Y', 'T240M', 'K211N'])\n",
    "SNP.drop(columns=['FID','PAT','MAT','PHENOTYPE','C253Y','K211N','SEX'], axis=1, inplace=True)\n",
    "SNP['WESTotal'] = SNP.drop('IID', axis=1).sum(axis=1)\n",
    "\n",
    "#Read in DEL1 csv file\n",
    "DEL1 = pd.read_csv('/path/to/del_list_1only.csv', na_values='NaN', header=0, names = ['IID','B', 'C', 'DEL1', 'DEL1_Exon', 'F'])\n",
    "DEL1.drop(columns=['B','C','F'], axis=1, inplace=True)\n",
    "\n",
    "#Read in DUP1 csv file\n",
    "DUP1 = pd.read_csv('/path/to/dup_list_1only.csv', na_values='NaN', header=0, names = ['IID','B', 'C', 'DUP1', 'DUP1_Exon', 'F'])\n",
    "DUP1.drop(columns=['B','C','F'], axis=1, inplace=True)\n",
    "\n",
    "#Read in DELDUP2 csv file\n",
    "DELDUP2 = pd.read_csv('/path/to/2DELDUP.csv', na_values='NaN', header=0)\n",
    "\n",
    "#Merge SNP and DEL\n",
    "df = pd.merge(SNP, DEL1, on ='IID', how ='left') \n",
    "df = pd.merge(df, DUP1, on ='IID', how ='left')\n",
    "df = pd.merge(df, DELDUP2, on ='IID', how ='left') \n",
    "\n",
    "#Fix na in in DEL1 and DUP1 columns\n",
    "df['DEL1'] = df['DEL1'].fillna(0)\n",
    "df['DEL1_Exon'] = df['DEL1_Exon'].fillna('')\n",
    "df['DUP1'] = df['DUP1'].fillna(0)\n",
    "df['DUP1_Exon'] = df['DUP1_Exon'].fillna('')\n",
    "df['2DELDUP'] = df['2DELDUP'].fillna(0)\n",
    "\n",
    "SNP['WESTotal'] = SNP.drop('IID', axis=1).sum(axis=1)\n",
    "\n",
    "df['CNVTotal'] = df['DEL1'] + df['DUP1'] + df['2DELDUP']\n",
    "df['MutTotal'] = df['CNVTotal'] + df['WESTotal']\n",
    "\n",
    "df\n",
    "\n",
    "dfcol = df.columns\n",
    "dfcol = dfcol[1:]\n",
    "\n",
    "COLUMN_NAMES = ['ID','AA_N','AA_f','AB_N','AB_f','BB_N','BB_f','ABorBB_N','ABorBB_f','Allelic_Frequency']\n",
    "results = pd.DataFrame(columns=COLUMN_NAMES)\n",
    "\n",
    "for x in dfcol:\n",
    "\n",
    "    test = df[x].value_counts()\n",
    "    \n",
    "    testlist = []\n",
    "    \n",
    "    testlist.append(x)\n",
    "\n",
    "    try:\n",
    "        a1 = test[0]\n",
    "        testlist.append(a1)\n",
    "        \n",
    "    except:\n",
    "        a1 = 0\n",
    "        testlist.append(a1)\n",
    "    \n",
    "    try:\n",
    "        a = test[0]/test.sum()\n",
    "        testlist.append(a)\n",
    "        \n",
    "    except:\n",
    "        a = 0\n",
    "        testlist.append(a)\n",
    "\n",
    "    try:\n",
    "        b1 =  test[1]\n",
    "        testlist.append(b1)\n",
    "            \n",
    "    except:\n",
    "        b1 =  0\n",
    "        testlist.append(b1)\n",
    "        \n",
    "    try:\n",
    "        b =  test[1]/test.sum()\n",
    "        testlist.append(b)\n",
    "            \n",
    "    except:\n",
    "        b =  0\n",
    "        testlist.append(b)\n",
    "\n",
    "    try:\n",
    "        c1 = test[2] \n",
    "        testlist.append(c1)\n",
    "            \n",
    "    except:\n",
    "        c1 = 0\n",
    "        testlist.append(c1)\n",
    "                \n",
    "    try:\n",
    "        c = test[2]/test.sum() \n",
    "        testlist.append(c)\n",
    "            \n",
    "    except:\n",
    "        c = 0\n",
    "        testlist.append(c)\n",
    "\n",
    "    try:\n",
    "         testlist.append((b1+c1))\n",
    "            \n",
    "    except:\n",
    "         testlist.append(0)\n",
    "\n",
    "    try:\n",
    "         testlist.append((b+c))\n",
    "            \n",
    "    except:\n",
    "         testlist.append(0)\n",
    "    \n",
    "    try:\n",
    "         testlist.append(((b/2)+c))\n",
    "            \n",
    "    except:\n",
    "         testlist.append(0) \n",
    "    \n",
    "    results_length = len(results)\n",
    "    \n",
    "    try:\n",
    "        results.loc[results_length] = testlist\n",
    "    except:\n",
    "        pass\n",
    "\n",
    "    \n",
    "#Write results to file\n",
    "results.to_csv('/path/to/Microarray_Frequency_Table.csv')\n",
    "results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Mut_Cases</th>\n",
       "      <th>Mut_CTRLs</th>\n",
       "      <th>NoMut_Cases</th>\n",
       "      <th>NoMut_CTRLs</th>\n",
       "      <th>Odds</th>\n",
       "      <th>p-value</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mutation</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>T240M</th>\n",
       "      <td>3</td>\n",
       "      <td>208</td>\n",
       "      <td>3455</td>\n",
       "      <td>484125</td>\n",
       "      <td>2.0210049538016253</td>\n",
       "      <td>0.18962020337317878</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R275W</th>\n",
       "      <td>34</td>\n",
       "      <td>3705</td>\n",
       "      <td>3389</td>\n",
       "      <td>476062</td>\n",
       "      <td>1.2890882584721786</td>\n",
       "      <td>0.1413468103455112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MicroarraySNP</th>\n",
       "      <td>37</td>\n",
       "      <td>3913</td>\n",
       "      <td>3428</td>\n",
       "      <td>480963</td>\n",
       "      <td>1.326669456835531</td>\n",
       "      <td>0.10423379705151295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DEL1</th>\n",
       "      <td>11</td>\n",
       "      <td>1422</td>\n",
       "      <td>3454</td>\n",
       "      <td>483454</td>\n",
       "      <td>1.0827443181309182</td>\n",
       "      <td>0.7511073348307502</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DUP1</th>\n",
       "      <td>7</td>\n",
       "      <td>1222</td>\n",
       "      <td>3458</td>\n",
       "      <td>483654</td>\n",
       "      <td>0.801192045959037</td>\n",
       "      <td>0.7322393651321926</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNV</th>\n",
       "      <td>18</td>\n",
       "      <td>2644</td>\n",
       "      <td>3447</td>\n",
       "      <td>482232</td>\n",
       "      <td>0.9524140573464527</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AllMut</th>\n",
       "      <td>55</td>\n",
       "      <td>6557</td>\n",
       "      <td>3410</td>\n",
       "      <td>478319</td>\n",
       "      <td>1.1765780967889525</td>\n",
       "      <td>0.23708875178743302</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Mut_Cases Mut_CTRLs NoMut_Cases NoMut_CTRLs                Odds  \\\n",
       "Mutation                                                                        \n",
       "T240M                 3       208        3455      484125  2.0210049538016253   \n",
       "R275W                34      3705        3389      476062  1.2890882584721786   \n",
       "MicroarraySNP        37      3913        3428      480963   1.326669456835531   \n",
       "DEL1                 11      1422        3454      483454  1.0827443181309182   \n",
       "DUP1                  7      1222        3458      483654   0.801192045959037   \n",
       "CNV                  18      2644        3447      482232  0.9524140573464527   \n",
       "AllMut               55      6557        3410      478319  1.1765780967889525   \n",
       "\n",
       "                           p-value  \n",
       "Mutation                            \n",
       "T240M          0.18962020337317878  \n",
       "R275W           0.1413468103455112  \n",
       "MicroarraySNP  0.10423379705151295  \n",
       "DEL1            0.7511073348307502  \n",
       "DUP1            0.7322393651321926  \n",
       "CNV                            1.0  \n",
       "AllMut         0.23708875178743302  "
      ]
     },
     "execution_count": 194,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#This code calculates the odds ratio and p-value for a single pathogenic or LOF PRKN mutations (detected by microarray only) in PD cases \n",
    "\n",
    "\n",
    "import pandas as pd\n",
    "import time\n",
    "import os\n",
    "import numpy as np\n",
    "import scipy.stats as stats\n",
    "\n",
    "\n",
    "#Read in SNP csv file\n",
    "SNP = pd.read_csv('/path/to/PRKN_pathogenic.csv', na_values='NaN', header=0, names = ['FID','IID', 'PAT', 'MAT', 'SEX', 'PHENOTYPE', 'R275W', 'C253Y', 'T240M', 'K211N'])\n",
    "SNP.drop(columns=['FID','PAT','MAT','PHENOTYPE','C253Y','K211N','SEX'], axis=1, inplace=True)\n",
    "SNP['WESTotal'] = SNP.drop('IID', axis=1).sum(axis=1)\n",
    "\n",
    "#Read in DEL1 csv file\n",
    "DEL1 = pd.read_csv('/path/to/del_list_1only.csv', na_values='NaN', header=0, names = ['IID','B', 'C', 'DEL1', 'DEL1_Exon', 'F'])\n",
    "DEL1.drop(columns=['B','C','F'], axis=1, inplace=True)\n",
    "\n",
    "#Read in DUP1 csv file\n",
    "DUP1 = pd.read_csv('/path/to/dup_list_1only.csv', na_values='NaN', header=0, names = ['IID','B', 'C', 'DUP1', 'DUP1_Exon', 'F'])\n",
    "DUP1.drop(columns=['B','C','F'], axis=1, inplace=True)\n",
    "\n",
    "#Read in DELDUP2 csv file\n",
    "DELDUP2 = pd.read_csv('/path/to/2DELDUP.csv', na_values='NaN', header=0)\n",
    "\n",
    "#Merge SNP and DEL\n",
    "df = pd.merge(SNP, DEL1, on ='IID', how ='left') \n",
    "df = pd.merge(df, DUP1, on ='IID', how ='left')\n",
    "df = pd.merge(df, DELDUP2, on ='IID', how ='left') \n",
    "\n",
    "#Fix na in in DEL1 and DUP1 columns\n",
    "df['DEL1'] = df['DEL1'].fillna(0)\n",
    "df['DEL1_Exon'] = df['DEL1_Exon'].fillna('')\n",
    "df['DUP1'] = df['DUP1'].fillna(0)\n",
    "df['DUP1_Exon'] = df['DUP1_Exon'].fillna('')\n",
    "df['2DELDUP'] = df['2DELDUP'].fillna(0)\n",
    "\n",
    "df['CNVTotal'] = df['DEL1'] + df['DUP1'] + df['2DELDUP']\n",
    "df['MutTotal'] = df['CNVTotal'] + df['WESTotal']\n",
    "\n",
    "#Read in PDStatus csv file\n",
    "PDStatus = pd.read_csv('/path/to/PD_April2021_with_covariates.csv', na_values='NaN', header=0)\n",
    "PDStatus['PD'] = 1\n",
    "\n",
    "FPDStatus = pd.read_csv('/path/to/ALL_indi_with_PD_parent_UNIQUE.csv', na_values='NaN', header=0)\n",
    "FPDStatus['FPD'] = 1\n",
    "\n",
    "#Merge SNP and DEL\n",
    "df2 = pd.merge(df, PDStatus, on ='IID', how ='left') \n",
    "df2 = pd.merge(df2, FPDStatus, on = 'IID', how='left')\n",
    "\n",
    "df2['PD'] = df2['PD'].fillna(0)\n",
    "df2['FPD'] = df2['FPD'].fillna(0)\n",
    "\n",
    "df2['PD'].value_counts()\n",
    "\n",
    "#Limit to cases with only 0 or 1 detected PRKN path or LOF variant\n",
    "df2 = df2.loc[df2['MutTotal'] <= 1.0]\n",
    "\n",
    "#PD in T240M\n",
    "PDT240M = df2.loc[(df2['T240M'] == 1) & (df2['PD'] == 1)]\n",
    "PDT240Mc = PDT240M['PD'].count()\n",
    "\n",
    "NoPDT240M = df2.loc[(df2['T240M'] == 1) & (df2['PD'] == 0)]\n",
    "NoPDT240Mc = NoPDT240M['PD'].count()\n",
    "\n",
    "PDNoT240M = df2.loc[(df2['T240M'] == 0) & (df2['PD'] == 1)]\n",
    "PDNoT240Mc = PDNoT240M['PD'].count()\n",
    "\n",
    "NoPDNoT240M = df2.loc[(df2['T240M'] == 0) & (df2['PD'] == 0)]\n",
    "NoPDNoT240Mc = NoPDNoT240M['PD'].count()\n",
    "\n",
    "testT240M, pvalueT240M = stats.fisher_exact([[PDT240Mc, NoPDT240Mc], [PDNoT240Mc, NoPDNoT240Mc]])\n",
    "\n",
    "#PD in R275W\n",
    "PDR275W = df2.loc[(df2['R275W'] == 1) & (df2['PD'] == 1)]\n",
    "PDR275Wc = PDR275W['PD'].count()\n",
    "\n",
    "NoPDR275W = df2.loc[(df2['R275W'] == 1) & (df2['PD'] == 0)]\n",
    "NoPDR275Wc = NoPDR275W['PD'].count()\n",
    "\n",
    "PDNoR275W = df2.loc[(df2['R275W'] == 0) & (df2['PD'] == 1)]\n",
    "PDNoR275Wc = PDNoR275W['PD'].count()\n",
    "\n",
    "NoPDNoR275W = df2.loc[(df2['R275W'] == 0) & (df2['PD'] == 0)]\n",
    "NoPDNoR275Wc = NoPDNoR275W['PD'].count()\n",
    "\n",
    "testR275W, pvalueR275W = stats.fisher_exact([[PDR275Wc, NoPDR275Wc], [PDNoR275Wc, NoPDNoR275Wc]])\n",
    "\n",
    "#PD in DEL1\n",
    "PDDEL1 = df2.loc[(df2['DEL1'] == 1) & (df2['PD'] == 1)]\n",
    "PDDEL1c = PDDEL1['PD'].count()\n",
    "\n",
    "NoPDDEL1 = df2.loc[(df2['DEL1'] == 1) & (df2['PD'] == 0)]\n",
    "NoPDDEL1c = NoPDDEL1['PD'].count()\n",
    "\n",
    "PDNoDEL1 = df2.loc[(df2['DEL1'] == 0) & (df2['PD'] == 1)]\n",
    "PDNoDEL1c = PDNoDEL1['PD'].count()\n",
    "\n",
    "NoPDNoDEL1 = df2.loc[(df2['DEL1'] == 0) & (df2['PD'] == 0)]\n",
    "NoPDNoDEL1c = NoPDNoDEL1['PD'].count()\n",
    "\n",
    "testDEL1, pvalueDEL1 = stats.fisher_exact([[PDDEL1c, NoPDDEL1c], [PDNoDEL1c, NoPDNoDEL1c]])\n",
    "\n",
    "#PD in DUP1\n",
    "PDDUP1 = df2.loc[(df2['DUP1'] == 1) & (df2['PD'] == 1)]\n",
    "PDDUP1c = PDDUP1['PD'].count()\n",
    "\n",
    "NoPDDUP1 = df2.loc[(df2['DUP1'] == 1) & (df2['PD'] == 0)]\n",
    "NoPDDUP1c = NoPDDUP1['PD'].count()\n",
    "\n",
    "PDNoDUP1 = df2.loc[(df2['DUP1'] == 0) & (df2['PD'] == 1)]\n",
    "PDNoDUP1c = PDNoDUP1['PD'].count()\n",
    "\n",
    "NoPDNoDUP1 = df2.loc[(df2['DUP1'] == 0) & (df2['PD'] == 0)]\n",
    "NoPDNoDUP1c = NoPDNoDUP1['PD'].count()\n",
    "\n",
    "testDUP1, pvalueDUP1 = stats.fisher_exact([[PDDUP1c, NoPDDUP1c], [PDNoDUP1c, NoPDNoDUP1c]])\n",
    "\n",
    "#PD in AnyCNV\n",
    "PDCNV = df2.loc[((df2['DUP1'] == 1) | (df2['DEL1'] == 1)) & (df2['PD'] == 1)]\n",
    "PDCNVc = PDCNV['PD'].count()\n",
    "\n",
    "NoPDCNV = df2.loc[((df2['DUP1'] == 1) | (df2['DEL1'] == 1)) & (df2['PD'] == 0)]\n",
    "NoPDCNVc = NoPDCNV['PD'].count()\n",
    "\n",
    "PDNoCNV = df2.loc[((df2['DUP1'] == 0) & (df2['DEL1'] == 0)) & (df2['PD'] == 1)]\n",
    "PDNoCNVc = PDNoCNV['PD'].count()\n",
    "\n",
    "NoPDNoCNV = df2.loc[((df2['DUP1'] == 0) & (df2['DEL1'] == 0)) & (df2['PD'] == 0)]\n",
    "NoPDNoCNVc = NoPDNoCNV['PD'].count()\n",
    "\n",
    "testCNV, pvalueCNV = stats.fisher_exact([[PDCNVc, NoPDCNVc], [PDNoCNVc, NoPDNoCNVc]])\n",
    "\n",
    "#PD in WESTotal\n",
    "PDWESTotal = df2.loc[(df2['WESTotal'] == 1) & (df2['PD'] == 1)]\n",
    "PDWESTotalc = PDWESTotal['PD'].count()\n",
    "\n",
    "NoPDWESTotal = df2.loc[(df2['WESTotal'] == 1) & (df2['PD'] == 0)]\n",
    "NoPDWESTotalc = NoPDWESTotal['PD'].count()\n",
    "\n",
    "PDNoWESTotal = df2.loc[(df2['WESTotal'] == 0) & (df2['PD'] == 1)]\n",
    "PDNoWESTotalc = PDNoWESTotal['PD'].count()\n",
    "\n",
    "NoPDNoWESTotal = df2.loc[(df2['WESTotal'] == 0) & (df2['PD'] == 0)]\n",
    "NoPDNoWESTotalc = NoPDNoWESTotal['PD'].count()\n",
    "\n",
    "testWESTotal, pvalueWESTotal = stats.fisher_exact([[PDWESTotalc, NoPDWESTotalc], [PDNoWESTotalc, NoPDNoWESTotalc]])\n",
    "\n",
    "\n",
    "#PD in MutTotal\n",
    "PDMutTotal = df2.loc[(df2['MutTotal'] == 1) & (df2['PD'] == 1)]\n",
    "PDMutTotalc = PDMutTotal['PD'].count()\n",
    "\n",
    "NoPDMutTotal = df2.loc[(df2['MutTotal'] == 1) & (df2['PD'] == 0)]\n",
    "NoPDMutTotalc = NoPDMutTotal['PD'].count()\n",
    "\n",
    "PDNoMutTotal = df2.loc[(df2['MutTotal'] == 0) & (df2['PD'] == 1)]\n",
    "PDNoMutTotalc = PDNoMutTotal['PD'].count()\n",
    "\n",
    "NoPDNoMutTotal = df2.loc[(df2['MutTotal'] == 0) & (df2['PD'] == 0)]\n",
    "NoPDNoMutTotalc = NoPDNoMutTotal['PD'].count()\n",
    "\n",
    "testMutTotal, pvalueMutTotal = stats.fisher_exact([[PDMutTotalc, NoPDMutTotalc], [PDNoMutTotalc, NoPDNoMutTotalc]])\n",
    "\n",
    "\n",
    "#Make pandas dataframe with result\n",
    "results = pd.DataFrame(np.array([['T240M', PDT240Mc, NoPDT240Mc, PDNoT240Mc, NoPDNoT240Mc, testT240M, pvalueT240M],['R275W', PDR275Wc, NoPDR275Wc, PDNoR275Wc, NoPDNoR275Wc, testR275W, pvalueR275W], ['MicroarraySNP', PDWESTotalc, NoPDWESTotalc, PDNoWESTotalc, NoPDNoWESTotalc, testWESTotal, pvalueWESTotal], ['DEL1', PDDEL1c, NoPDDEL1c, PDNoDEL1c, NoPDNoDEL1c, testDEL1, pvalueDEL1], ['DUP1', PDDUP1c, NoPDDUP1c, PDNoDUP1c, NoPDNoDUP1c, testDUP1, pvalueDUP1],['CNV', PDCNVc, NoPDCNVc, PDNoCNVc, NoPDNoCNVc, testCNV, pvalueCNV],['AllMut', PDMutTotalc, NoPDMutTotalc, PDNoMutTotalc, NoPDNoMutTotalc, testMutTotal, pvalueMutTotal]]), columns=['Mutation','Mut_Cases', 'Mut_CTRLs', 'NoMut_Cases','NoMut_CTRLs','Odds','p-value'])\n",
    "results.set_index('Mutation', inplace=True)\n",
    "\n",
    "#Write results to file\n",
    "results.to_csv('/path/to/MicroarrayMutation_PD_Odds_Table.csv')\n",
    "results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Mut_Cases</th>\n",
       "      <th>Mut_CTRLs</th>\n",
       "      <th>NoMut_Cases</th>\n",
       "      <th>NoMut_CTRLs</th>\n",
       "      <th>Odds</th>\n",
       "      <th>p-value</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mutation</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>T240M</th>\n",
       "      <td>9</td>\n",
       "      <td>202</td>\n",
       "      <td>17644</td>\n",
       "      <td>469936</td>\n",
       "      <td>1.186677770021391</td>\n",
       "      <td>0.5775949679519168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R275W</th>\n",
       "      <td>135</td>\n",
       "      <td>3604</td>\n",
       "      <td>17363</td>\n",
       "      <td>462088</td>\n",
       "      <td>0.9968938376175038</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MicroarraySNP</th>\n",
       "      <td>144</td>\n",
       "      <td>3806</td>\n",
       "      <td>17531</td>\n",
       "      <td>466860</td>\n",
       "      <td>1.0075664179657666</td>\n",
       "      <td>0.9318019186661882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DEL1</th>\n",
       "      <td>65</td>\n",
       "      <td>1368</td>\n",
       "      <td>17610</td>\n",
       "      <td>469298</td>\n",
       "      <td>1.2662416855122853</td>\n",
       "      <td>0.0654163270876398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DUP1</th>\n",
       "      <td>43</td>\n",
       "      <td>1186</td>\n",
       "      <td>17632</td>\n",
       "      <td>469480</td>\n",
       "      <td>0.9653821964051258</td>\n",
       "      <td>0.8785908906394373</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNV</th>\n",
       "      <td>108</td>\n",
       "      <td>2554</td>\n",
       "      <td>17567</td>\n",
       "      <td>468112</td>\n",
       "      <td>1.1268212685572663</td>\n",
       "      <td>0.2310247294794069</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AllMut</th>\n",
       "      <td>252</td>\n",
       "      <td>6360</td>\n",
       "      <td>17423</td>\n",
       "      <td>464306</td>\n",
       "      <td>1.055904849261278</td>\n",
       "      <td>0.3888563232544414</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Mut_Cases Mut_CTRLs NoMut_Cases NoMut_CTRLs                Odds  \\\n",
       "Mutation                                                                        \n",
       "T240M                 9       202       17644      469936   1.186677770021391   \n",
       "R275W               135      3604       17363      462088  0.9968938376175038   \n",
       "MicroarraySNP       144      3806       17531      466860  1.0075664179657666   \n",
       "DEL1                 65      1368       17610      469298  1.2662416855122853   \n",
       "DUP1                 43      1186       17632      469480  0.9653821964051258   \n",
       "CNV                 108      2554       17567      468112  1.1268212685572663   \n",
       "AllMut              252      6360       17423      464306   1.055904849261278   \n",
       "\n",
       "                          p-value  \n",
       "Mutation                           \n",
       "T240M          0.5775949679519168  \n",
       "R275W                         1.0  \n",
       "MicroarraySNP  0.9318019186661882  \n",
       "DEL1           0.0654163270876398  \n",
       "DUP1           0.8785908906394373  \n",
       "CNV            0.2310247294794069  \n",
       "AllMut         0.3888563232544414  "
      ]
     },
     "execution_count": 195,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#This code calculates the odds ratio and p-value for a single pathogenic or LOF PRKN mutations (detected by microarray only) in participants with or without a parent with PD\n",
    "\n",
    "\n",
    "import pandas as pd\n",
    "import time\n",
    "import os\n",
    "import numpy as np\n",
    "import scipy.stats as stats\n",
    "\n",
    "\n",
    "#Read in SNP csv file\n",
    "SNP = pd.read_csv('/path/to/PRKN_pathogenic.csv', na_values='NaN', header=0, names = ['FID','IID', 'PAT', 'MAT', 'SEX', 'PHENOTYPE', 'R275W', 'C253Y', 'T240M', 'K211N'])\n",
    "SNP.drop(columns=['FID','PAT','MAT','PHENOTYPE','C253Y','K211N','SEX'], axis=1, inplace=True)\n",
    "SNP['WESTotal'] = SNP.drop('IID', axis=1).sum(axis=1)\n",
    "\n",
    "#Read in DEL1 csv file\n",
    "DEL1 = pd.read_csv('/path/to/del_list_1only.csv', na_values='NaN', header=0, names = ['IID','B', 'C', 'DEL1', 'DEL1_Exon', 'F'])\n",
    "DEL1.drop(columns=['B','C','F'], axis=1, inplace=True)\n",
    "\n",
    "#Read in DUP1 csv file\n",
    "DUP1 = pd.read_csv('/path/to/dup_list_1only.csv', na_values='NaN', header=0, names = ['IID','B', 'C', 'DUP1', 'DUP1_Exon', 'F'])\n",
    "DUP1.drop(columns=['B','C','F'], axis=1, inplace=True)\n",
    "\n",
    "#Read in DELDUP2 csv file\n",
    "DELDUP2 = pd.read_csv('/path/to/2DELDUP.csv', na_values='NaN', header=0)\n",
    "\n",
    "#Merge SNP and DEL\n",
    "df = pd.merge(SNP, DEL1, on ='IID', how ='left') \n",
    "df = pd.merge(df, DUP1, on ='IID', how ='left')\n",
    "df = pd.merge(df, DELDUP2, on ='IID', how ='left') \n",
    "\n",
    "#Fix na in in DEL1 and DUP1 columns\n",
    "df['DEL1'] = df['DEL1'].fillna(0)\n",
    "df['DEL1_Exon'] = df['DEL1_Exon'].fillna('')\n",
    "df['DUP1'] = df['DUP1'].fillna(0)\n",
    "df['DUP1_Exon'] = df['DUP1_Exon'].fillna('')\n",
    "df['2DELDUP'] = df['2DELDUP'].fillna(0)\n",
    "\n",
    "df['CNVTotal'] = df['DEL1'] + df['DUP1'] + df['2DELDUP']\n",
    "df['MutTotal'] = df['CNVTotal'] + df['WESTotal']\n",
    "\n",
    "#Read in PDStatus csv file\n",
    "PDStatus = pd.read_csv('/path/to/PD_April2021_with_covariates.csv', na_values='NaN', header=0)\n",
    "PDStatus['FPD'] = 1\n",
    "\n",
    "FPDStatus = pd.read_csv('/path/to/ALL_indi_with_PD_parent_UNIQUE.csv', na_values='NaN', header=0)\n",
    "FPDStatus['PD'] = 1\n",
    "\n",
    "#Merge SNP and DEL\n",
    "df2 = pd.merge(df, PDStatus, on ='IID', how ='left') \n",
    "df2 = pd.merge(df2, FPDStatus, on = 'IID', how='left')\n",
    "\n",
    "df2['PD'] = df2['PD'].fillna(0)\n",
    "df2['FPD'] = df2['FPD'].fillna(0)\n",
    "\n",
    "df2['PD'].value_counts()\n",
    "\n",
    "#Limit to cases with only 0 or 1 detected PRKN path or LOF variant\n",
    "df2 = df2.loc[df2['MutTotal'] <= 1.0]\n",
    "\n",
    "#PD in T240M\n",
    "PDT240M = df2.loc[(df2['T240M'] == 1) & (df2['PD'] == 1)]\n",
    "PDT240Mc = PDT240M['PD'].count()\n",
    "\n",
    "NoPDT240M = df2.loc[(df2['T240M'] == 1) & (df2['PD'] == 0)]\n",
    "NoPDT240Mc = NoPDT240M['PD'].count()\n",
    "\n",
    "PDNoT240M = df2.loc[(df2['T240M'] == 0) & (df2['PD'] == 1)]\n",
    "PDNoT240Mc = PDNoT240M['PD'].count()\n",
    "\n",
    "NoPDNoT240M = df2.loc[(df2['T240M'] == 0) & (df2['PD'] == 0)]\n",
    "NoPDNoT240Mc = NoPDNoT240M['PD'].count()\n",
    "\n",
    "testT240M, pvalueT240M = stats.fisher_exact([[PDT240Mc, NoPDT240Mc], [PDNoT240Mc, NoPDNoT240Mc]])\n",
    "\n",
    "#PD in R275W\n",
    "PDR275W = df2.loc[(df2['R275W'] == 1) & (df2['PD'] == 1)]\n",
    "PDR275Wc = PDR275W['PD'].count()\n",
    "\n",
    "NoPDR275W = df2.loc[(df2['R275W'] == 1) & (df2['PD'] == 0)]\n",
    "NoPDR275Wc = NoPDR275W['PD'].count()\n",
    "\n",
    "PDNoR275W = df2.loc[(df2['R275W'] == 0) & (df2['PD'] == 1)]\n",
    "PDNoR275Wc = PDNoR275W['PD'].count()\n",
    "\n",
    "NoPDNoR275W = df2.loc[(df2['R275W'] == 0) & (df2['PD'] == 0)]\n",
    "NoPDNoR275Wc = NoPDNoR275W['PD'].count()\n",
    "\n",
    "testR275W, pvalueR275W = stats.fisher_exact([[PDR275Wc, NoPDR275Wc], [PDNoR275Wc, NoPDNoR275Wc]])\n",
    "\n",
    "#PD in DEL1\n",
    "PDDEL1 = df2.loc[(df2['DEL1'] == 1) & (df2['PD'] == 1)]\n",
    "PDDEL1c = PDDEL1['PD'].count()\n",
    "\n",
    "NoPDDEL1 = df2.loc[(df2['DEL1'] == 1) & (df2['PD'] == 0)]\n",
    "NoPDDEL1c = NoPDDEL1['PD'].count()\n",
    "\n",
    "PDNoDEL1 = df2.loc[(df2['DEL1'] == 0) & (df2['PD'] == 1)]\n",
    "PDNoDEL1c = PDNoDEL1['PD'].count()\n",
    "\n",
    "NoPDNoDEL1 = df2.loc[(df2['DEL1'] == 0) & (df2['PD'] == 0)]\n",
    "NoPDNoDEL1c = NoPDNoDEL1['PD'].count()\n",
    "\n",
    "testDEL1, pvalueDEL1 = stats.fisher_exact([[PDDEL1c, NoPDDEL1c], [PDNoDEL1c, NoPDNoDEL1c]])\n",
    "\n",
    "#PD in DUP1\n",
    "PDDUP1 = df2.loc[(df2['DUP1'] == 1) & (df2['PD'] == 1)]\n",
    "PDDUP1c = PDDUP1['PD'].count()\n",
    "\n",
    "NoPDDUP1 = df2.loc[(df2['DUP1'] == 1) & (df2['PD'] == 0)]\n",
    "NoPDDUP1c = NoPDDUP1['PD'].count()\n",
    "\n",
    "PDNoDUP1 = df2.loc[(df2['DUP1'] == 0) & (df2['PD'] == 1)]\n",
    "PDNoDUP1c = PDNoDUP1['PD'].count()\n",
    "\n",
    "NoPDNoDUP1 = df2.loc[(df2['DUP1'] == 0) & (df2['PD'] == 0)]\n",
    "NoPDNoDUP1c = NoPDNoDUP1['PD'].count()\n",
    "\n",
    "testDUP1, pvalueDUP1 = stats.fisher_exact([[PDDUP1c, NoPDDUP1c], [PDNoDUP1c, NoPDNoDUP1c]])\n",
    "\n",
    "#PD in AnyCNV\n",
    "PDCNV = df2.loc[((df2['DUP1'] == 1) | (df2['DEL1'] == 1)) & (df2['PD'] == 1)]\n",
    "PDCNVc = PDCNV['PD'].count()\n",
    "\n",
    "NoPDCNV = df2.loc[((df2['DUP1'] == 1) | (df2['DEL1'] == 1)) & (df2['PD'] == 0)]\n",
    "NoPDCNVc = NoPDCNV['PD'].count()\n",
    "\n",
    "PDNoCNV = df2.loc[((df2['DUP1'] == 0) & (df2['DEL1'] == 0)) & (df2['PD'] == 1)]\n",
    "PDNoCNVc = PDNoCNV['PD'].count()\n",
    "\n",
    "NoPDNoCNV = df2.loc[((df2['DUP1'] == 0) & (df2['DEL1'] == 0)) & (df2['PD'] == 0)]\n",
    "NoPDNoCNVc = NoPDNoCNV['PD'].count()\n",
    "\n",
    "testCNV, pvalueCNV = stats.fisher_exact([[PDCNVc, NoPDCNVc], [PDNoCNVc, NoPDNoCNVc]])\n",
    "\n",
    "#PD in WESTotal\n",
    "PDWESTotal = df2.loc[(df2['WESTotal'] == 1) & (df2['PD'] == 1)]\n",
    "PDWESTotalc = PDWESTotal['PD'].count()\n",
    "\n",
    "NoPDWESTotal = df2.loc[(df2['WESTotal'] == 1) & (df2['PD'] == 0)]\n",
    "NoPDWESTotalc = NoPDWESTotal['PD'].count()\n",
    "\n",
    "PDNoWESTotal = df2.loc[(df2['WESTotal'] == 0) & (df2['PD'] == 1)]\n",
    "PDNoWESTotalc = PDNoWESTotal['PD'].count()\n",
    "\n",
    "NoPDNoWESTotal = df2.loc[(df2['WESTotal'] == 0) & (df2['PD'] == 0)]\n",
    "NoPDNoWESTotalc = NoPDNoWESTotal['PD'].count()\n",
    "\n",
    "testWESTotal, pvalueWESTotal = stats.fisher_exact([[PDWESTotalc, NoPDWESTotalc], [PDNoWESTotalc, NoPDNoWESTotalc]])\n",
    "\n",
    "\n",
    "#PD in MutTotal\n",
    "PDMutTotal = df2.loc[(df2['MutTotal'] == 1) & (df2['PD'] == 1)]\n",
    "PDMutTotalc = PDMutTotal['PD'].count()\n",
    "\n",
    "NoPDMutTotal = df2.loc[(df2['MutTotal'] == 1) & (df2['PD'] == 0)]\n",
    "NoPDMutTotalc = NoPDMutTotal['PD'].count()\n",
    "\n",
    "PDNoMutTotal = df2.loc[(df2['MutTotal'] == 0) & (df2['PD'] == 1)]\n",
    "PDNoMutTotalc = PDNoMutTotal['PD'].count()\n",
    "\n",
    "NoPDNoMutTotal = df2.loc[(df2['MutTotal'] == 0) & (df2['PD'] == 0)]\n",
    "NoPDNoMutTotalc = NoPDNoMutTotal['PD'].count()\n",
    "\n",
    "testMutTotal, pvalueMutTotal = stats.fisher_exact([[PDMutTotalc, NoPDMutTotalc], [PDNoMutTotalc, NoPDNoMutTotalc]])\n",
    "\n",
    "\n",
    "#Make pandas dataframe with result\n",
    "results = pd.DataFrame(np.array([['T240M', PDT240Mc, NoPDT240Mc, PDNoT240Mc, NoPDNoT240Mc, testT240M, pvalueT240M],['R275W', PDR275Wc, NoPDR275Wc, PDNoR275Wc, NoPDNoR275Wc, testR275W, pvalueR275W], ['MicroarraySNP', PDWESTotalc, NoPDWESTotalc, PDNoWESTotalc, NoPDNoWESTotalc, testWESTotal, pvalueWESTotal], ['DEL1', PDDEL1c, NoPDDEL1c, PDNoDEL1c, NoPDNoDEL1c, testDEL1, pvalueDEL1], ['DUP1', PDDUP1c, NoPDDUP1c, PDNoDUP1c, NoPDNoDUP1c, testDUP1, pvalueDUP1],['CNV', PDCNVc, NoPDCNVc, PDNoCNVc, NoPDNoCNVc, testCNV, pvalueCNV],['AllMut', PDMutTotalc, NoPDMutTotalc, PDNoMutTotalc, NoPDNoMutTotalc, testMutTotal, pvalueMutTotal]]), columns=['Mutation','Mut_Cases', 'Mut_CTRLs', 'NoMut_Cases','NoMut_CTRLs','Odds','p-value'])\n",
    "results.set_index('Mutation', inplace=True)\n",
    "\n",
    "#Write results to file\n",
    "results.to_csv('/path/to/MicroarrayMutation_FPD_Odds_Table.csv')\n",
    "results\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of mutations detected by microarray in biallelic PRKN participants:\n",
      "2.0    13\n",
      "1.0     8\n",
      "0.0     3\n",
      "Name: microTotal, dtype: int64\n",
      "Number of mutations detected by WES + microarray in biallelic PRKN participants:\n",
      "2.0    24\n",
      "Name: MutTotal, dtype: int64\n",
      "Number of mutations detected by microarray in 1 and 2 PRKN mutation carriers:\n",
      "1.0    2679\n",
      "0.0     954\n",
      "2.0      13\n",
      "Name: microTotal, dtype: int64\n",
      "Number of mutations detected by WES + microarray in 1 and 2 PRKN mutation carriers:\n",
      "1.0    3671\n",
      "2.0      24\n",
      "Name: MutTotal, dtype: int64\n",
      "Proportion of PRKN mutations detected by microarray:\n",
      "0.7294971766603926\n",
      "Proportion of PRKN SNPs detected by microarray:\n",
      "0.42350094111320247\n"
     ]
    }
   ],
   "source": [
    "#This code determines what proportion of mutations are identified by microarray using WES SNPs + microarray CNVs as gold standard\n",
    "\n",
    "import pandas as pd\n",
    "import time\n",
    "import os\n",
    "import numpy as np\n",
    "import scipy.stats as stats\n",
    "\n",
    "\n",
    "#Read in SNP csv file\n",
    "SNP = pd.read_csv('/path/to/PRKN_variants_WES_touse.csv', na_values='NaN', header=0)\n",
    "SNP.drop(columns = \"number_of_var\", inplace=True)\n",
    "SNP['WESTotal'] = SNP.drop('IID', axis=1).sum(axis=1)\n",
    "\n",
    "#Read in DEL1 csv file\n",
    "DEL1 = pd.read_csv('/path/to/del_list_1only.csv', na_values='NaN', header=0, names = ['IID','B', 'C', 'DEL1', 'DEL1_Exon', 'F'])\n",
    "DEL1.drop(columns=['B','C','F'], axis=1, inplace=True)\n",
    "\n",
    "#Read in DUP1 csv file\n",
    "DUP1 = pd.read_csv('/path/to/dup_list_1only.csv', na_values='NaN', header=0, names = ['IID','B', 'C', 'DUP1', 'DUP1_Exon', 'F'])\n",
    "DUP1.drop(columns=['B','C','F'], axis=1, inplace=True)\n",
    "\n",
    "#Read in DELDUP2 csv file\n",
    "DELDUP2 = pd.read_csv('/path/to/2DELDUP.csv', na_values='NaN', header=0)\n",
    "\n",
    "#Read in mSNP csv file\n",
    "mSNP = pd.read_csv('/path/to/PRKN_pathogenic.csv', na_values='NaN', header=0, names = ['FID','IID', 'PAT', 'MAT', 'SEX', 'PHENOTYPE', 'R275W', 'C253Y', 'T240M', 'K211N'])\n",
    "mSNP.drop(columns=['FID','PAT','MAT','PHENOTYPE','C253Y','K211N','SEX'], axis=1, inplace=True)\n",
    "mSNP['mSNPTotal'] = mSNP.drop('IID', axis=1).sum(axis=1)\n",
    "\n",
    "\n",
    "#Read in PDStatus csv file\n",
    "PDStatus = pd.read_csv('/path/to/PD_April2021_with_covariates.csv', na_values='NaN', header=0)\n",
    "PDStatus['FPD'] = 1\n",
    "\n",
    "FPDStatus = pd.read_csv('/path/to/ALL_indi_with_PD_parent_UNIQUE.csv', na_values='NaN', header=0)\n",
    "FPDStatus['PD'] = 1\n",
    "\n",
    "#Merge SNP and DEL\n",
    "df2 = pd.merge(df, PDStatus, on ='IID', how ='left') \n",
    "df2 = pd.merge(df2, FPDStatus, on = 'IID', how='left')\n",
    "\n",
    "#Merge SNP and DEL\n",
    "df = pd.merge(SNP, DEL1, on ='IID', how ='left') \n",
    "df = pd.merge(df, DUP1, on ='IID', how ='left')\n",
    "df = pd.merge(df, DELDUP2, on ='IID', how ='left')\n",
    "df = pd.merge(df, mSNP, on ='IID', how ='left')\n",
    "df = pd.merge(df, PDStatus, on ='IID', how ='left') \n",
    "df = pd.merge(df, FPDStatus, on = 'IID', how='left')\n",
    "\n",
    "\n",
    "#Fix na in in DEL1 and DUP1 columns\n",
    "df['DEL1'] = df['DEL1'].fillna(0)\n",
    "df['DEL1_Exon'] = df['DEL1_Exon'].fillna('')\n",
    "df['DUP1'] = df['DUP1'].fillna(0)\n",
    "df['DUP1_Exon'] = df['DUP1_Exon'].fillna('')\n",
    "df['2DELDUP'] = df['2DELDUP'].fillna(0)\n",
    "df['PD'] = df['PD'].fillna(0)\n",
    "df['FPD'] = df['FPD'].fillna(0)\n",
    "\n",
    "#Read in key to annotate\n",
    "WESkey = pd.read_csv('/path/to/WESKey.csv', na_values='NaN', header=0)\n",
    "\n",
    "df['CNVTotal'] = df['DEL1'] + df['DUP1'] + df['2DELDUP']\n",
    "df['MutTotal'] = df['CNVTotal'] + df['WESTotal']\n",
    "df['microTotal'] = df['DEL1'] + df['DUP1'] + df['2DELDUP'] + df['R275W'] + df['T240M'] \n",
    "\n",
    "twomuts = df.loc[df['MutTotal'] >= 2]\n",
    "anymut = df.loc[df['MutTotal'] >= 1]\n",
    "\n",
    "print('Number of mutations detected by microarray in biallelic PRKN participants:')\n",
    "print(twomuts['microTotal'].value_counts())\n",
    "print('Number of mutations detected by WES + microarray in biallelic PRKN participants:')\n",
    "print(twomuts['MutTotal'].value_counts())\n",
    "print('Number of mutations detected by microarray in 1 and 2 PRKN mutation carriers:')\n",
    "print(anymut['microTotal'].value_counts())\n",
    "print('Number of mutations detected by WES + microarray in 1 and 2 PRKN mutation carriers:')\n",
    "print(anymut['MutTotal'].value_counts())\n",
    "\n",
    "posmicro = anymut[(anymut['microTotal']>=1)]\n",
    "negmicro = anymut[(anymut['microTotal']==0)]\n",
    "SNPmicro = anymut[(anymut['R275W']>=1) | (anymut['T240M']>=1)]\n",
    "\n",
    "print('Proportion of PRKN mutations detected by microarray:')\n",
    "print(posmicro['MutTotal'].sum() / anymut['MutTotal'].sum())\n",
    "print('Proportion of PRKN SNPs detected by microarray:')\n",
    "print(SNPmicro['microTotal'].sum() / anymut['MutTotal'].sum())\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
